From d0b6de27d18e5245904144b1b3a4737d019df75c Mon Sep 17 00:00:00 2001
From: Kiwi Browser <40272275+kiwibrowser@users.noreply.github.com>
Date: Fri, 13 Aug 2021 03:57:56 +0300
Subject: [PATCH 293/506] Additional FB Messenger improvements

---
 .../chrome/browser/PersonalizeResults.java        | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/PersonalizeResults.java b/chrome/android/java/src/org/chromium/chrome/browser/PersonalizeResults.java
index 7a91ed2d..417fc5c0 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/PersonalizeResults.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/PersonalizeResults.java
@@ -28,6 +28,9 @@ public class PersonalizeResults {
                        || tab.getUrl().getSpec().startsWith("https://m.facebook.com/messages"))) {
           tab.getWebContents().evaluateJavaScript(MESSENGER_SCRIPT, null);
        }
+       if (tab != null && tab.getUrl().getSpec().contains("messenger.com/")) {
+          tab.getWebContents().evaluateJavaScript(MESSENGER_VIEWPORT_SCRIPT, null);
+       }
        if (tab != null && tab.getUrl().getSpec().startsWith("https://m.facebook.com/")) {
           tab.getWebContents().evaluateJavaScript("(function(){ if (!document.location.href.includes('https://m.facebook.com/')) { return; } document.querySelector('body.touch').style = \"cursor:default\";})();", null);
        }
@@ -116,11 +119,23 @@ public class PersonalizeResults {
 
     private static final String MESSENGER_SCRIPT = ""
 +"(function() {"
++"window.addEventListener('load', function() {"
 +"if (!document.location.href.includes('https://m.facebook.com/messenger/install') && !document.location.href.includes('https://m.facebook.com/messages')) { return; } "
 +"var gotomessenger = document.createElement('div');"
 +"gotomessenger.innerHTML = \"<a href='https://www.messenger.com' target='_blank' style='margin: 2rem; display: inline-block;'><b>Go to www.messenger.com instead</a>\";"
++"gotomessenger.id = '_kb_gotomessenger';"
 +"var e1 = document.querySelector('._8rws') || document.querySelector('._2bu8');"
++"if (!document.getElementById('_kb_gotomessenger'))"
 +"e1.parentNode.insertBefore(gotomessenger, e1.nextSibling);"
++"});"
++"})();";
+
+    private static final String MESSENGER_VIEWPORT_SCRIPT = ""
++"(function() {"
++"if (!document.location.href.includes('messenger.com/')) { return; } "
++"var sheet = document.createElement('style');"
++"sheet.innerHTML = '.jgljxmt5 { min-height: calc(93vh - var(--header-height)); } .g0mhvs5p.g0mhvs5p { width: 150px; }';"
++"document.body.appendChild(sheet);"
 +"})();";
 
     private static final String AMP_SCRIPT = ""
