diff --git a/chrome/browser/media/router/BUILD.gn b/chrome/browser/media/router/BUILD.gn
index d0c9384251b8f..d63a06ef4e1ce 100644
--- a/chrome/browser/media/router/BUILD.gn
+++ b/chrome/browser/media/router/BUILD.gn
@@ -416,7 +416,7 @@ source_set("unittests") {
   }
 
   # Traditional desktop platforms and desktop android with extensions support.
-  if (!is_android || enable_desktop_android_extensions) {
+  if (!is_android || is_desktop_android) {
     deps += [ "//chrome/browser/media/router/discovery:test_support" ]
   }
 
diff --git a/extensions/common/extension_features.cc b/extensions/common/extension_features.cc
index 8c3c00819fc3e..301318e701873 100644
--- a/extensions/common/extension_features.cc
+++ b/extensions/common/extension_features.cc
@@ -114,12 +114,12 @@ BASE_FEATURE(kExtensionLocalizationGuid, base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kExtensionIconVariants, base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kExtensionManifestV2Unsupported, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kExtensionManifestV2Unsupported, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kExtensionManifestV2ExceptionList,
              base::FEATURE_DISABLED_BY_DEFAULT);
 
-BASE_FEATURE(kExtensionManifestV2Disabled, base::FEATURE_ENABLED_BY_DEFAULT);
+BASE_FEATURE(kExtensionManifestV2Disabled, base::FEATURE_DISABLED_BY_DEFAULT);
 
 BASE_FEATURE(kExtensionsBackgroundCompilation,
              base::FEATURE_DISABLED_BY_DEFAULT);
diff --git a/ui/webui/resources/cr_components/cr_shortcut_input/BUILD.gn b/ui/webui/resources/cr_components/cr_shortcut_input/BUILD.gn
index 778452c32838f..a8c29b2d966ea 100644
--- a/ui/webui/resources/cr_components/cr_shortcut_input/BUILD.gn
+++ b/ui/webui/resources/cr_components/cr_shortcut_input/BUILD.gn
@@ -4,8 +4,9 @@
 
 import("//ui/webui/resources/tools/build_webui.gni")
 import("//ui/webui/webui_features.gni")
+import("//extensions/buildflags/buildflags.gni")
 
-assert((!is_android && !is_ios) || enable_webui_ntp || is_desktop_android)
+assert((!is_android && !is_ios) || enable_webui_ntp || is_desktop_android || enable_desktop_android_extensions)
 
 build_webui("build") {
   grd_prefix = "cr_components_cr_shortcut_input"
diff --git a/ui/webui/resources/cr_components/managed_footnote/BUILD.gn b/ui/webui/resources/cr_components/managed_footnote/BUILD.gn
index 98582036e1624..78e59258f62cc 100644
--- a/ui/webui/resources/cr_components/managed_footnote/BUILD.gn
+++ b/ui/webui/resources/cr_components/managed_footnote/BUILD.gn
@@ -5,7 +5,8 @@
 import("//ui/webui/resources/tools/build_webui.gni")
 import("//ui/webui/webui_features.gni")
 
-assert((!is_android && !is_ios) || enable_webui_ntp || is_desktop_android)
+import("//extensions/buildflags/buildflags.gni")
+assert((!is_android && !is_ios) || enable_webui_ntp || is_desktop_android || enable_desktop_android_extensions)
 
 build_webui("build") {
   grd_prefix = "cr_components_managed_footnote"
