diff --git a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
index 17f7465c79133..b7d5efa29693c 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/toolbar/ToolbarManager.java
@@ -2638,7 +2638,7 @@ public class ToolbarManager
                                                         mActivityTabProvider.asObservable(),
                                                         mTabCreatorManager.getTabCreator(false),
                                                         getBrowsingModeThemeColorProvider(),
-                                                        (ToolbarTablet) mToolbarLayout,
+                                                        mToolbarLayout,
                                                         contextMenuPopulatorFactory,
                                                         selectionDropdownMenuDelegate,
                                                         mTabModelSelector,
diff --git a/chrome/browser/ui/android/toolbar/java/res/layout/toolbar_phone.xml b/chrome/browser/ui/android/toolbar/java/res/layout/toolbar_phone.xml
index 9e1f21d5a0c8e..eec8fddb248c2 100644
--- a/chrome/browser/ui/android/toolbar/java/res/layout/toolbar_phone.xml
+++ b/chrome/browser/ui/android/toolbar/java/res/layout/toolbar_phone.xml
@@ -74,6 +74,12 @@ found in the LICENSE file.
             android:layout_height="match_parent"
             android:visibility="gone"/>
 
+        <ViewStub
+            android:id="@+id/extensions_toolbar_container_stub"
+            android:inflatedId="@+id/extensions_toolbar_container"
+            android:layout_width="wrap_content"
+            android:layout_height="match_parent"
+            android:layout_gravity="top"/>
         <org.chromium.chrome.browser.toolbar.top.ToggleTabStackButton
             android:id="@+id/tab_switcher_button"
             style="@style/ToolbarHoverableButton"
