diff --git a/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuHandlerImpl.java b/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuHandlerImpl.java
index 965d17e3..b4fa2f11 100644
--- a/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuHandlerImpl.java
+++ b/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenuHandlerImpl.java
@@ -1,4 +1,4 @@
-// Copyright 2013 The Chromium Authors. All rights reserved.
+// Copyright 2013 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -20,13 +20,18 @@ import androidx.annotation.VisibleForTesting;
 import org.chromium.base.Callback;
 import org.chromium.base.metrics.RecordUserAction;
 import org.chromium.base.supplier.Supplier;
+import org.chromium.chrome.browser.browser_controls.BrowserControlsStateProvider;
 import org.chromium.chrome.browser.lifecycle.ActivityLifecycleDispatcher;
 import org.chromium.chrome.browser.lifecycle.ConfigurationChangedObserver;
 import org.chromium.chrome.browser.lifecycle.StartStopWithNativeObserver;
 import org.chromium.chrome.browser.ui.appmenu.internal.R;
 import org.chromium.components.browser_ui.widget.textbubble.TextBubble;
+import org.chromium.ui.KeyboardVisibilityDelegate;
+import org.chromium.ui.base.WindowAndroid;
 import org.chromium.ui.display.DisplayAndroidManager;
 import org.chromium.ui.modelutil.LayoutViewBuilder;
+import org.chromium.ui.modelutil.ListObservable;
+import org.chromium.ui.modelutil.ListObservable.ListObserver;
 import org.chromium.ui.modelutil.MVCListAdapter.ModelList;
 import org.chromium.ui.modelutil.ModelListAdapter;
 import org.chromium.ui.modelutil.PropertyModel;
@@ -54,35 +59,47 @@ class AppMenuHandlerImpl
     private final View mDecorView;
     private final ActivityLifecycleDispatcher mActivityLifecycleDispatcher;
     private final Supplier<Rect> mAppRect;
-
+    private final WindowAndroid mWindowAndroid;
+    private final BrowserControlsStateProvider mBrowserControlsStateProvider;
+    private ModelList mModelList;
+    private ListObserver<Void> mListObserver;
     private Callback<Integer> mTestOptionsItemSelectedListener;
+    private KeyboardVisibilityDelegate.KeyboardVisibilityListener mKeyboardVisibilityListener;
 
     /**
-     * The resource id of the menu item to highlight when the menu next opens. A value of
-     * {@code null} means no item will be highlighted.  This value will be cleared after the menu is
-     * opened.
+     * The resource id of the menu item to highlight when the menu next opens. A value of {@code
+     * null} means no item will be highlighted. This value will be cleared after the menu is opened.
      */
     private Integer mHighlightMenuId;
 
     /**
      * Constructs an AppMenuHandlerImpl object.
+     *
      * @param context The activity context.
      * @param delegate Delegate used to check the desired AppMenu properties on show.
      * @param appMenuDelegate The AppMenuDelegate to handle menu item selection.
-     * @param menuResourceId Resource Id that should be used as the source for the menu items.
-     *            It is assumed to have back_menu_id, forward_menu_id, bookmark_this_page_id.
+     * @param menuResourceId Resource Id that should be used as the source for the menu items. It is
+     *     assumed to have back_menu_id, forward_menu_id, bookmark_this_page_id.
      * @param decorView The decor {@link View}, e.g. from Window#getDecorView(), for the containing
-     *            activity.
+     *     activity.
      * @param activityLifecycleDispatcher The {@link ActivityLifecycleDispatcher} for the containing
-     *            activity.
+     *     activity.
      * @param hardwareButtonAnchorView The {@link View} used as an anchor for the menu when it is
-     *            displayed using a hardware button.
+     *     displayed using a hardware button.
      * @param appRect Supplier of the app area in Window that the menu should fit in.
+     * @param windowAndroid The window that will be used to fetch {@link KeyboardVisibilityDelegate}
+     * @param browserControlsStateProvider a provider that can provide the state of the toolbar
      */
-    public AppMenuHandlerImpl(Context context, AppMenuPropertiesDelegate delegate,
-            AppMenuDelegate appMenuDelegate, View decorView,
-            ActivityLifecycleDispatcher activityLifecycleDispatcher, View hardwareButtonAnchorView,
-            Supplier<Rect> appRect) {
+    public AppMenuHandlerImpl(
+            Context context,
+            AppMenuPropertiesDelegate delegate,
+            AppMenuDelegate appMenuDelegate,
+            View decorView,
+            ActivityLifecycleDispatcher activityLifecycleDispatcher,
+            View hardwareButtonAnchorView,
+            Supplier<Rect> appRect,
+            WindowAndroid windowAndroid,
+            BrowserControlsStateProvider browserControlsStateProvider) {
         mContext = context;
         mAppMenuDelegate = appMenuDelegate;
         mDelegate = delegate;
@@ -91,19 +108,46 @@ class AppMenuHandlerImpl
         mObservers = new ArrayList<>();
         mHardwareButtonMenuAnchor = hardwareButtonAnchorView;
         mAppRect = appRect;
+        mWindowAndroid = windowAndroid;
+        mBrowserControlsStateProvider = browserControlsStateProvider;
 
         mActivityLifecycleDispatcher = activityLifecycleDispatcher;
         mActivityLifecycleDispatcher.register(this);
 
-        assert mHardwareButtonMenuAnchor
-                != null : "Using AppMenu requires to have menu_anchor_stub view";
+        assert mHardwareButtonMenuAnchor != null
+                : "Using AppMenu requires to have menu_anchor_stub view";
+        mListObserver =
+                new ListObserver<Void>() {
+                    @Override
+                    public void onItemRangeInserted(ListObservable source, int index, int count) {
+                        assert mModelList != null && mAppMenu != null;
+                        updateModelForHighlightAndClick(
+                                mModelList,
+                                mHighlightMenuId,
+                                mAppMenu,
+                                /* startIndex= */ index,
+                                /* withAssertions= */ false);
+                    }
+
+                    @Override
+                    public void onItemRangeRemoved(ListObservable source, int index, int count) {
+                        assert mModelList != null;
+                        updateModelForHighlightAndClick(
+                                mModelList,
+                                mHighlightMenuId,
+                                mAppMenu,
+                                /* startIndex= */ index,
+                                /* withAssertions= */ false);
+                    }
+                };
     }
 
-    /**
-     * Called when the containing activity is being destroyed.
-     */
+    /** Called when the containing activity is being destroyed. */
     void destroy() {
         // Prevent the menu window from leaking.
+        mWindowAndroid
+                .getKeyboardDelegate()
+                .removeKeyboardVisibilityListener(mKeyboardVisibilityListener);
         hideAppMenu();
 
         mActivityLifecycleDispatcher.unregister(this);
@@ -121,27 +165,33 @@ class AppMenuHandlerImpl
 
     @Override
     public void setMenuHighlight(Integer highlightItemId) {
+        boolean highlighting = highlightItemId != null;
+        setMenuHighlight(highlightItemId, highlighting);
+    }
+
+    @Override
+    public void setMenuHighlight(Integer highlightItemId, boolean shouldHighlightMenuButton) {
         if (mHighlightMenuId == null && highlightItemId == null) return;
         if (mHighlightMenuId != null && mHighlightMenuId.equals(highlightItemId)) return;
         mHighlightMenuId = highlightItemId;
-        boolean highlighting = mHighlightMenuId != null;
-        for (AppMenuObserver observer : mObservers) observer.onMenuHighlightChanged(highlighting);
+        for (AppMenuObserver observer : mObservers) {
+            observer.onMenuHighlightChanged(shouldHighlightMenuButton);
+        }
     }
 
     /**
      * Show the app menu.
-     * @param anchorView    Anchor view (usually a menu button) to be used for the popup, if null is
-     *                      passed then hardware menu button anchor will be used.
+     *
+     * @param anchorView Anchor view (usually a menu button) to be used for the popup, if null is
+     *     passed then hardware menu button anchor will be used.
      * @param startDragging Whether dragging is started. For example, if the app menu is showed by
-     *                      tapping on a button, this should be false. If it is showed by start
-     *                      dragging down on the menu button, this should be true. Note that if
-     *                      anchorView is null, this must be false since we no longer support
-     *                      hardware menu button dragging.
-     * @return              True, if the menu is shown, false, if menu is not shown, example
-     *                      reasons: the menu is not yet available to be shown, or the menu is
-     *                      already showing.
+     *     tapping on a button, this should be false. If it is showed by start dragging down on the
+     *     menu button, this should be true. Note that if anchorView is null, this must be false
+     *     since we no longer support hardware menu button dragging.
+     * @return True, if the menu is shown, false, if menu is not shown, example reasons: the menu is
+     *     not yet available to be shown, or the menu is already showing.
      */
-    // TODO(crbug.com/635567): Fix this properly.
+    // TODO(crbug.com/40479664): Fix this properly.
     @SuppressLint("ResourceType")
     boolean showAppMenu(View anchorView, boolean startDragging) {
         if (!shouldShowAppMenu() || isAppMenuShowing()) return false;
@@ -167,7 +217,8 @@ class AppMenuHandlerImpl
         // If the anchor view used to show the popup or the activity's decor view is not attached
         // to window, we don't show the app menu because the window manager might have revoked
         // the window token for this activity. See https://crbug.com/1105831.
-        if (!mDecorView.isAttachedToWindow() || !anchorView.isAttachedToWindow()
+        if (!mDecorView.isAttachedToWindow()
+                || !anchorView.isAttachedToWindow()
                 || !anchorView.getRootView().isAttachedToWindow()) {
             return false;
         }
@@ -177,55 +228,72 @@ class AppMenuHandlerImpl
         List<CustomViewBinder> customViewBinders = mDelegate.getCustomViewBinders();
         Map<CustomViewBinder, Integer> customViewTypeOffsetMap =
                 populateCustomViewBinderOffsetMap(customViewBinders, AppMenuItemType.NUM_ENTRIES);
-        ModelList modelList = mDelegate.getMenuItems(((id) -> {
-            return getCustomItemViewType(id, customViewBinders, customViewTypeOffsetMap);
-        }),
-                this);
-
+        mModelList =
+                mDelegate.getMenuItems(
+                        (id) -> {
+                            return getCustomItemViewType(
+                                    id, customViewBinders, customViewTypeOffsetMap);
+                        },
+                        this);
+        mModelList.addObserver(mListObserver);
         ContextThemeWrapper wrapper =
                 new ContextThemeWrapper(mContext, R.style.OverflowMenuThemeOverlay);
 
         if (mAppMenu == null) {
-            TypedArray a = wrapper.obtainStyledAttributes(
-                    new int[] {android.R.attr.listPreferredItemHeightSmall});
+            TypedArray a =
+                    wrapper.obtainStyledAttributes(
+                            new int[] {android.R.attr.listPreferredItemHeightSmall});
             int itemRowHeight = a.getDimensionPixelSize(0, 0);
             a.recycle();
             mAppMenu = new AppMenu(itemRowHeight, this, mContext.getResources());
             mAppMenuDragHelper = new AppMenuDragHelper(mContext, mAppMenu, itemRowHeight);
         }
-        setupModelForHighlightAndClick(modelList, mHighlightMenuId, mAppMenu);
-        ModelListAdapter adapter = new ModelListAdapter(modelList);
-        mAppMenu.updateMenu(modelList, adapter);
-        registerViewBinders(customViewBinders, customViewTypeOffsetMap, adapter,
+        setupModelForHighlightAndClick(mModelList, mHighlightMenuId, mAppMenu);
+        ModelListAdapter adapter = new ModelListAdapter(mModelList);
+        mAppMenu.updateMenu(mModelList, adapter);
+        registerViewBinders(
+                customViewBinders,
+                customViewTypeOffsetMap,
+                adapter,
                 mDelegate.shouldShowIconBeforeItem());
 
-        Rect appRect = mAppRect.get();
-
-        // Use full size of window for abnormal appRect.
-        if (appRect.left < 0 && appRect.top < 0) {
-            appRect.left = 0;
-            appRect.top = 0;
-            appRect.right = mDecorView.getWidth();
-            appRect.bottom = mDecorView.getHeight();
-        }
-
         Point pt = new Point();
         display.getSize(pt);
 
-        int footerResourceId = 0;
-        if (mDelegate.shouldShowFooter(appRect.height())) {
-            footerResourceId = mDelegate.getFooterResourceId();
+        KeyboardVisibilityDelegate keyboardVisibilityDelegate =
+                mWindowAndroid.getKeyboardDelegate();
+
+        // If keyboard is showing, wait until keyboard disappears to set appRect
+        if (keyboardVisibilityDelegate.isKeyboardShowing(mContext, anchorView)) {
+            View finalAnchorView = anchorView;
+            boolean finalIsByPermanentButton = isByPermanentButton;
+            mKeyboardVisibilityListener =
+                    isShowing -> {
+                        if (!isShowing) {
+                            setDisplayAndShowAppMenu(
+                                    wrapper,
+                                    finalAnchorView,
+                                    finalIsByPermanentButton,
+                                    rotation,
+                                    mAppRect.get(),
+                                    customViewBinders,
+                                    startDragging);
+                            keyboardVisibilityDelegate
+                                    .removeKeyboardVisibilityListener(mKeyboardVisibilityListener);
+                        }
+                    };
+            keyboardVisibilityDelegate.addKeyboardVisibilityListener(mKeyboardVisibilityListener);
+            keyboardVisibilityDelegate.hideKeyboard(anchorView);
+        } else {
+            setDisplayAndShowAppMenu(
+                    wrapper,
+                    anchorView,
+                    isByPermanentButton,
+                    rotation,
+                    mAppRect.get(),
+                    customViewBinders,
+                    startDragging);
         }
-        int headerResourceId = 0;
-        if (mDelegate.shouldShowHeader(appRect.height())) {
-            headerResourceId = mDelegate.getHeaderResourceId();
-        }
-        mAppMenu.show(wrapper, anchorView, isByPermanentButton, rotation, appRect, pt.y,
-                footerResourceId, headerResourceId, mDelegate.getGroupDividerId(), mHighlightMenuId,
-                customViewBinders, mDelegate.isMenuIconAtStart());
-        mAppMenuDragHelper.onShow(startDragging);
-        clearMenuHighlight();
-        RecordUserAction.record("MobileMenuShow");
         return true;
     }
 
@@ -252,7 +320,12 @@ class AppMenuHandlerImpl
 
     @Override
     public void hideAppMenu() {
-        if (mAppMenu != null && mAppMenu.isShowing()) mAppMenu.dismiss();
+        if (mAppMenu != null && mAppMenu.isShowing()) {
+            mAppMenu.dismiss();
+            if (mModelList != null) {
+                mModelList.removeObserver(mListObserver);
+            }
+        }
     }
 
     @Override
@@ -290,14 +363,13 @@ class AppMenuHandlerImpl
     }
 
     @VisibleForTesting
-    void onOptionsItemSelected(int itemId, boolean highlighted) {
+    void onOptionsItemSelected(int itemId) {
         if (mTestOptionsItemSelectedListener != null) {
             mTestOptionsItemSelectedListener.onResult(itemId);
             return;
         }
 
         mAppMenuDelegate.onOptionsItemSelected(itemId, mDelegate.getBundleForMenuItem(itemId));
-        if (highlighted) mDelegate.recordHighlightedMenuItemClicked(itemId);
     }
 
     /**
@@ -348,37 +420,43 @@ class AppMenuHandlerImpl
         return true;
     }
 
-    @VisibleForTesting
     void overrideOnOptionItemSelectedListenerForTests(
             Callback<Integer> onOptionsItemSelectedListener) {
         mTestOptionsItemSelectedListener = onOptionsItemSelectedListener;
     }
 
-    @VisibleForTesting
     AppMenuPropertiesDelegate getDelegateForTests() {
         return mDelegate;
     }
 
-    private void registerViewBinders(@Nullable List<CustomViewBinder> customViewBinders,
-            Map<CustomViewBinder, Integer> customViewTypeOffsetMap, ModelListAdapter adapter,
+    private void registerViewBinders(
+            @Nullable List<CustomViewBinder> customViewBinders,
+            Map<CustomViewBinder, Integer> customViewTypeOffsetMap,
+            ModelListAdapter adapter,
             boolean iconBeforeItem) {
         int standardItemResId = R.layout.menu_item;
         if (iconBeforeItem) {
             standardItemResId = R.layout.menu_item_start_with_icon;
         }
 
-        adapter.registerType(AppMenuItemType.STANDARD, new LayoutViewBuilder(standardItemResId),
+        adapter.registerType(
+                AppMenuItemType.STANDARD,
+                new LayoutViewBuilder(standardItemResId),
                 AppMenuItemViewBinder::bindStandardItem);
-        adapter.registerType(AppMenuItemType.TITLE_BUTTON,
+        adapter.registerType(
+                AppMenuItemType.TITLE_BUTTON,
                 new LayoutViewBuilder(R.layout.title_button_menu_item),
                 AppMenuItemViewBinder::bindTitleButtonItem);
-        adapter.registerType(AppMenuItemType.THREE_BUTTON_ROW,
+        adapter.registerType(
+                AppMenuItemType.THREE_BUTTON_ROW,
                 new LayoutViewBuilder(R.layout.icon_row_menu_item),
                 AppMenuItemViewBinder::bindIconRowItem);
-        adapter.registerType(AppMenuItemType.FOUR_BUTTON_ROW,
+        adapter.registerType(
+                AppMenuItemType.FOUR_BUTTON_ROW,
                 new LayoutViewBuilder(R.layout.icon_row_menu_item),
                 AppMenuItemViewBinder::bindIconRowItem);
-        adapter.registerType(AppMenuItemType.FIVE_BUTTON_ROW,
+        adapter.registerType(
+                AppMenuItemType.FIVE_BUTTON_ROW,
                 new LayoutViewBuilder(R.layout.icon_row_menu_item),
                 AppMenuItemViewBinder::bindIconRowItem);
 
@@ -390,36 +468,56 @@ class AppMenuHandlerImpl
             }
 
             for (int j = 0; j < binder.getViewTypeCount(); j++) {
-                adapter.registerType(customViewTypeOffsetMap.get(binder) + j,
-                        new LayoutViewBuilder(binder.getLayoutId(j)), binder);
+                adapter.registerType(
+                        customViewTypeOffsetMap.get(binder) + j,
+                        new LayoutViewBuilder(binder.getLayoutId(j)),
+                        binder);
             }
         }
     }
 
     void setupModelForHighlightAndClick(
             ModelList modelList, Integer highlightedId, AppMenuClickHandler appMenuClickHandler) {
+        updateModelForHighlightAndClick(
+                modelList,
+                highlightedId,
+                appMenuClickHandler,
+                /* startIndex= */ 0,
+                /* withAssertions= */ true);
+    }
+
+    private void updateModelForHighlightAndClick(
+            ModelList modelList,
+            Integer highlightedId,
+            AppMenuClickHandler appMenuClickHandler,
+            int startIndex,
+            boolean withAssertions) {
         if (modelList == null) {
             return;
         }
 
-        for (int i = 0; i < modelList.size(); i++) {
+        for (int i = startIndex; i < modelList.size(); i++) {
             PropertyModel model = modelList.get(i).model;
-            // Not like other keys which is set by AppMenuPropertiesDelegateImpl, CLICK_HANDLER and
-            // HIGHLIGHTED should not be set yet.
-            assert model.get(AppMenuItemProperties.CLICK_HANDLER) == null;
-            assert !model.get(AppMenuItemProperties.HIGHLIGHTED);
+            if (withAssertions) {
+                // Not like other keys which is set by AppMenuPropertiesDelegateImpl, CLICK_HANDLER
+                // and HIGHLIGHTED should not be set yet.
+                assert model.get(AppMenuItemProperties.CLICK_HANDLER) == null;
+                assert !model.get(AppMenuItemProperties.HIGHLIGHTED);
+            }
             model.set(AppMenuItemProperties.CLICK_HANDLER, appMenuClickHandler);
             model.set(AppMenuItemProperties.POSITION, i);
 
             if (highlightedId != null) {
-                model.set(AppMenuItemProperties.HIGHLIGHTED,
+                model.set(
+                        AppMenuItemProperties.HIGHLIGHTED,
                         model.get(AppMenuItemProperties.MENU_ITEM_ID) == highlightedId);
                 if (model.get(AppMenuItemProperties.SUBMENU) != null) {
                     ModelList subList = model.get(AppMenuItemProperties.SUBMENU);
                     for (int j = 0; j < subList.size(); j++) {
                         PropertyModel subModel = subList.get(j).model;
                         subModel.set(AppMenuItemProperties.CLICK_HANDLER, appMenuClickHandler);
-                        subModel.set(AppMenuItemProperties.HIGHLIGHTED,
+                        subModel.set(
+                                AppMenuItemProperties.HIGHLIGHTED,
                                 subModel.get(AppMenuItemProperties.MENU_ITEM_ID) == highlightedId);
                     }
                 }
@@ -441,7 +539,9 @@ class AppMenuHandlerImpl
         return customViewTypeOffsetMap;
     }
 
-    private int getCustomItemViewType(int id, List<CustomViewBinder> customViewBinders,
+    private int getCustomItemViewType(
+            int id,
+            List<CustomViewBinder> customViewBinders,
             Map<CustomViewBinder, Integer> customViewTypeOffsetMap) {
         if (customViewBinders == null || customViewTypeOffsetMap == null) {
             return CustomViewBinder.NOT_HANDLED;
@@ -461,4 +561,56 @@ class AppMenuHandlerImpl
     static void setExceptionReporter(Callback<Throwable> reporter) {
         AppMenu.setExceptionReporter(reporter);
     }
+
+    @Nullable
+    ModelList getModelListForTesting() {
+        return mModelList;
+    }
+
+    public View getKeyboardDelegate() {
+        return mDecorView;
+    }
+
+    private void setDisplayAndShowAppMenu(
+            ContextThemeWrapper wrapper,
+            View anchorView,
+            boolean isByPermanentButton,
+            Integer rotation,
+            Rect appRect,
+            List<CustomViewBinder> customViewBinders,
+            boolean startDragging) {
+        // Use full size of window for abnormal appRect.
+        if (appRect.left < 0 && appRect.top < 0) {
+            appRect.left = 0;
+            appRect.top = 0;
+            appRect.right = mDecorView.getWidth();
+            appRect.bottom = mDecorView.getHeight();
+        }
+
+        int footerResourceId = 0;
+        if (mDelegate.shouldShowFooter(appRect.height())) {
+            footerResourceId = mDelegate.getFooterResourceId();
+        }
+        int headerResourceId = 0;
+        if (mDelegate.shouldShowHeader(appRect.height())) {
+            headerResourceId = mDelegate.getHeaderResourceId();
+        }
+        mAppMenu.show(
+                wrapper,
+                anchorView,
+                isByPermanentButton,
+                rotation,
+                appRect,
+                footerResourceId,
+                headerResourceId,
+                mDelegate.getGroupDividerId(),
+                mHighlightMenuId,
+                customViewBinders,
+                mDelegate.isMenuIconAtStart(),
+                mBrowserControlsStateProvider.getControlsPosition());
+        mAppMenuDragHelper.onShow(startDragging);
+        clearMenuHighlight();
+        RecordUserAction.record("MobileMenuShow");
+        mDelegate.onMenuShown();
+    }
 }
