diff --git a/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenu.java b/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenu.java
index b42c2c36..cb99c614 100644
--- a/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenu.java
+++ b/chrome/browser/ui/android/appmenu/internal/java/src/org/chromium/chrome/browser/ui/appmenu/AppMenu.java
@@ -1,4 +1,4 @@
-// Copyright 2011 The Chromium Authors. All rights reserved.
+// Copyright 2011 The Chromium Authors
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
@@ -12,7 +12,6 @@ import android.graphics.Color;
 import android.graphics.Rect;
 import android.graphics.drawable.ColorDrawable;
 import android.graphics.drawable.Drawable;
-import android.os.Build;
 import android.os.SystemClock;
 import android.text.TextUtils;
 import android.view.Gravity;
@@ -42,6 +41,7 @@ import org.chromium.base.SysUtils;
 import org.chromium.base.metrics.RecordHistogram;
 import org.chromium.base.task.PostTask;
 import org.chromium.base.task.TaskTraits;
+import org.chromium.chrome.browser.browser_controls.BrowserControlsStateProvider.ControlsPosition;
 import org.chromium.chrome.browser.ui.appmenu.internal.R;
 import org.chromium.components.browser_ui.styles.ChromeColors;
 import org.chromium.components.browser_ui.widget.chips.ChipView;
@@ -71,7 +71,6 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
     private final int mItemRowHeight;
     private final int mVerticalFadeDistance;
     private final int mNegativeSoftwareVerticalOffset;
-    private final int mNegativeVerticalOffsetNotTopAnchored;
     private final int mChipHighlightExtension;
     private final int[] mTempLocation;
 
@@ -85,7 +84,6 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
     private AnimatorSet mMenuItemEnterAnimator;
     private long mMenuShownTimeMs;
     private boolean mSelectedItemBeforeDismiss;
-    private Integer mHighlightedItemId;
     private ModelList mModelList;
 
     /**
@@ -103,8 +101,6 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         mNegativeSoftwareVerticalOffset =
                 res.getDimensionPixelSize(R.dimen.menu_negative_software_vertical_offset);
         mVerticalFadeDistance = res.getDimensionPixelSize(R.dimen.menu_vertical_fade_distance);
-        mNegativeVerticalOffsetNotTopAnchored =
-                res.getDimensionPixelSize(R.dimen.menu_negative_vertical_offset_not_top_anchored);
         mChipHighlightExtension =
                 res.getDimensionPixelOffset(R.dimen.menu_chip_highlight_extension);
 
@@ -113,10 +109,10 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
 
     /**
      * Notifies the menu that the contents of the menu item specified by {@code menuRowId} have
-     * changed.  This should be called if icons, titles, etc. are changing for a particular menu
-     * item while the menu is open.
-     * @param menuRowId The id of the menu item to change.  This must be a row id and not a child
-     *                  id.
+     * changed. This should be called if icons, titles, etc. are changing for a particular menu item
+     * while the menu is open.
+     *
+     * @param menuRowId The id of the menu item to change. This must be a row id and not a child id.
      */
     public void menuItemContentChanged(int menuRowId) {
         // Make sure we have all the valid state objects we need.
@@ -151,63 +147,64 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
     /**
      * Creates and shows the app menu anchored to the specified view.
      *
-     * @param context               The context of the AppMenu (ensure the proper theme is set on
-     *                              this context).
-     * @param anchorView            The anchor {@link View} of the {@link PopupWindow}.
-     * @param isByPermanentButton   Whether or not permanent hardware button triggered it. (oppose
-     *                              to software button or keyboard).
-     * @param screenRotation        Current device screen rotation.
-     * @param visibleDisplayFrame   The display area rect in which AppMenu is supposed to fit in.
-     * @param screenHeight          Current device screen height.
-     * @param footerResourceId      The resource id for a view to add as a fixed view at the bottom
-     *                              of the menu.  Can be 0 if no such view is required.  The footer
-     *                              is always visible and overlays other app menu items if
-     *                              necessary.
-     * @param headerResourceId      The resource id for a view to add as the first item in menu
-     *                              list. Can be null if no such view is required. See
-     *                              {@link ListView#addHeaderView(View)}.
-     * @param highlightedItemId     The resource id of the menu item that should be highlighted.
-     *                              Can be {@code null} if no item should be highlighted.  Note that
-     *                              {@code 0} is dedicated to custom menu items and can be declared
-     *                              by external apps.
-     * @param groupDividerResourceId     The resource id of divider menu items. This will be used to
-     *         determine the number of dividers that appear in the menu.
-     * @param customViewBinders     See {@link AppMenuPropertiesDelegate#getCustomViewBinders()}.
-     * @param isMenuIconAtStart     Whether the menu is being shown from a menu icon positioned at
-     *                              the start.
+     * @param context The context of the AppMenu (ensure the proper theme is set on this context).
+     * @param anchorView The anchor {@link View} of the {@link PopupWindow}.
+     * @param isByPermanentButton Whether or not permanent hardware button triggered it. (oppose to
+     *     software button or keyboard).
+     * @param screenRotation Current device screen rotation.
+     * @param visibleDisplayFrame The display area rect in which AppMenu is supposed to fit in.
+     * @param footerResourceId The resource id for a view to add as a fixed view at the bottom of
+     *     the menu. Can be 0 if no such view is required. The footer is always visible and overlays
+     *     other app menu items if necessary.
+     * @param headerResourceId The resource id for a view to add as the first item in menu list. Can
+     *     be null if no such view is required. See {@link ListView#addHeaderView(View)}.
+     * @param highlightedItemId The resource id of the menu item that should be highlighted. Can be
+     *     {@code null} if no item should be highlighted. Note that {@code 0} is dedicated to custom
+     *     menu items and can be declared by external apps.
+     * @param groupDividerResourceId The resource id of divider menu items. This will be used to
+     *     determine the number of dividers that appear in the menu.
+     * @param customViewBinders See {@link AppMenuPropertiesDelegate#getCustomViewBinders()}.
+     * @param isMenuIconAtStart Whether the menu is being shown from a menu icon positioned at the
+     *     start.
      */
-    void show(Context context, final View anchorView, boolean isByPermanentButton,
-            int screenRotation, Rect visibleDisplayFrame, int screenHeight,
-            @IdRes int footerResourceId, @IdRes int headerResourceId,
-            @IdRes int groupDividerResourceId, Integer highlightedItemId,
-            @Nullable List<CustomViewBinder> customViewBinders, boolean isMenuIconAtStart) {
+    void show(
+            Context context,
+            final View anchorView,
+            boolean isByPermanentButton,
+            int screenRotation,
+            Rect visibleDisplayFrame,
+            @IdRes int footerResourceId,
+            @IdRes int headerResourceId,
+            @IdRes int groupDividerResourceId,
+            Integer highlightedItemId,
+            @Nullable List<CustomViewBinder> customViewBinders,
+            boolean isMenuIconAtStart,
+            @ControlsPosition int controlsPosition) {
         mPopup = new PopupWindow(context);
         mPopup.setFocusable(true);
         mPopup.setInputMethodMode(PopupWindow.INPUT_METHOD_NOT_NEEDED);
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-            // The window layout type affects the z-index of the popup window on M+.
-            mPopup.setWindowLayoutType(WindowManager.LayoutParams.TYPE_APPLICATION_SUB_PANEL);
-        }
+        // The window layout type affects the z-index of the popup window.
+        mPopup.setWindowLayoutType(WindowManager.LayoutParams.TYPE_APPLICATION_SUB_PANEL);
 
-        mPopup.setOnDismissListener(() -> {
-            recordTimeToTakeActionHistogram();
-            if (anchorView instanceof ImageButton) {
-                ((ImageButton) anchorView).setSelected(false);
-            }
+        mPopup.setOnDismissListener(
+                () -> {
+                    recordTimeToTakeActionHistogram();
+                    if (anchorView instanceof ImageButton) {
+                        ((ImageButton) anchorView).setSelected(false);
+                    }
 
-            if (mMenuItemEnterAnimator != null) mMenuItemEnterAnimator.cancel();
+                    if (mMenuItemEnterAnimator != null) mMenuItemEnterAnimator.cancel();
 
-            mHandler.appMenuDismissed();
-            mHandler.onMenuVisibilityChanged(false);
+                    mHandler.appMenuDismissed();
+                    mHandler.onMenuVisibilityChanged(false);
 
-            mPopup = null;
-            mAdapter = null;
-            mListView = null;
-            mFooterView = null;
-            mMenuItemEnterAnimator = null;
-            mHighlightedItemId = null;
-        });
+                    mPopup = null;
+                    mAdapter = null;
+                    mListView = null;
+                    mFooterView = null;
+                    mMenuItemEnterAnimator = null;
+                });
 
         // Some OEMs don't actually let us change the background... but they still return the
         // padding of the new background, which breaks the menu height.  If we still have a
@@ -223,7 +220,11 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
 
         if (!isByPermanentButton) {
             mPopup.setAnimationStyle(
-                    isMenuIconAtStart ? R.style.StartIconMenuAnim : R.style.EndIconMenuAnim);
+                    isMenuIconAtStart
+                            ? R.style.StartIconMenuAnim
+                            : (controlsPosition == ControlsPosition.TOP
+                                    ? R.style.EndIconMenuAnim
+                                    : R.style.EndIconMenuAnimBottom));
         }
 
         // Turn off window animations for low end devices.
@@ -267,7 +268,6 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         int footerHeight = inflateFooter(footerResourceId, contentView, menuWidth);
         int headerHeight = inflateHeader(headerResourceId, contentView, menuWidth);
 
-        mHighlightedItemId = highlightedItemId;
         if (highlightedItemId != null) {
             View viewToHighlight = contentView.findViewById(highlightedItemId);
             HighlightParams highlightParams = new HighlightParams(HighlightShape.RECTANGLE);
@@ -291,20 +291,39 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         // See crbug.com/761726.
         mListView.setAdapter(mAdapter);
 
-        int popupHeight = setMenuHeight(menuItemIds, heightList, visibleDisplayFrame, screenHeight,
-                sizingPadding, footerHeight, headerHeight, anchorView, groupDividerResourceId);
-        int[] popupPosition = getPopupPosition(mTempLocation, mIsByPermanentButton,
-                mNegativeSoftwareVerticalOffset, mNegativeVerticalOffsetNotTopAnchored,
-                mCurrentScreenRotation, visibleDisplayFrame, sizingPadding, anchorView, popupWidth,
-                popupHeight, anchorView.getRootView().getLayoutDirection());
+        anchorView.getLocationOnScreen(mTempLocation);
+        int anchorViewOffset =
+                Math.min(
+                        Math.abs(mTempLocation[1] - visibleDisplayFrame.top),
+                        Math.abs(mTempLocation[1] - visibleDisplayFrame.bottom));
+        setMenuHeight(
+                menuItemIds,
+                heightList,
+                visibleDisplayFrame,
+                sizingPadding,
+                footerHeight,
+                headerHeight,
+                anchorView,
+                groupDividerResourceId,
+                anchorViewOffset);
+        int[] popupPosition =
+                getPopupPosition(
+                        mTempLocation,
+                        mIsByPermanentButton,
+                        mNegativeSoftwareVerticalOffset,
+                        mCurrentScreenRotation,
+                        visibleDisplayFrame,
+                        sizingPadding,
+                        anchorView,
+                        popupWidth,
+                        anchorView.getRootView().getLayoutDirection());
         mPopup.setContentView(contentView);
 
-        if (popupHeight + popupPosition[1] > visibleDisplayFrame.bottom) {
-            mPopup.setHeight(visibleDisplayFrame.height());
-        }
-
         try {
-            mPopup.showAtLocation(anchorView.getRootView(), Gravity.NO_GRAVITY, popupPosition[0],
+            mPopup.showAtLocation(
+                    anchorView.getRootView(),
+                    Gravity.NO_GRAVITY,
+                    popupPosition[0],
                     popupPosition[1]);
         } catch (WindowManager.BadTokenException e) {
             // Intentionally ignore BadTokenException. This can happen in a real edge case where
@@ -329,22 +348,37 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
 
         // Don't animate the menu items for low end devices.
         if (!SysUtils.isLowEndDevice()) {
-            mListView.addOnLayoutChangeListener(new View.OnLayoutChangeListener() {
-                @Override
-                public void onLayoutChange(View v, int left, int top, int right, int bottom,
-                        int oldLeft, int oldTop, int oldRight, int oldBottom) {
-                    mListView.removeOnLayoutChangeListener(this);
-                    runMenuItemEnterAnimations();
-                }
-            });
+            mListView.addOnLayoutChangeListener(
+                    new View.OnLayoutChangeListener() {
+                        @Override
+                        public void onLayoutChange(
+                                View v,
+                                int left,
+                                int top,
+                                int right,
+                                int bottom,
+                                int oldLeft,
+                                int oldTop,
+                                int oldRight,
+                                int oldBottom) {
+                            mListView.removeOnLayoutChangeListener(this);
+                            runMenuItemEnterAnimations();
+                        }
+                    });
         }
     }
 
     @VisibleForTesting
-    static int[] getPopupPosition(int[] tempLocation, boolean isByPermanentButton,
-            int negativeSoftwareVerticalOffset, int negativeVerticalOffsetNotTopAnchored,
-            int screenRotation, Rect appRect, Rect padding, View anchorView, int popupWidth,
-            int popupHeight, int viewLayoutDirection) {
+    static int[] getPopupPosition(
+            int[] tempLocation,
+            boolean isByPermanentButton,
+            int negativeSoftwareVerticalOffset,
+            int screenRotation,
+            Rect appRect,
+            Rect padding,
+            View anchorView,
+            int popupWidth,
+            int viewLayoutDirection) {
         anchorView.getLocationInWindow(tempLocation);
         int anchorViewX = tempLocation[0];
         int anchorViewY = tempLocation[1];
@@ -392,7 +426,7 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         int id = model.get(AppMenuItemProperties.MENU_ITEM_ID);
         mSelectedItemBeforeDismiss = true;
         dismiss();
-        mHandler.onOptionsItemSelected(id, mHighlightedItemId != null && mHighlightedItemId == id);
+        mHandler.onOptionsItemSelected(id);
     }
 
     @Override
@@ -401,9 +435,10 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
 
         mSelectedItemBeforeDismiss = true;
         CharSequence titleCondensed = model.get(AppMenuItemProperties.TITLE_CONDENSED);
-        CharSequence message = TextUtils.isEmpty(titleCondensed)
-                ? model.get(AppMenuItemProperties.TITLE)
-                : titleCondensed;
+        CharSequence message =
+                TextUtils.isEmpty(titleCondensed)
+                        ? model.get(AppMenuItemProperties.TITLE)
+                        : titleCondensed;
         return showToastForItem(message, view);
     }
 
@@ -454,9 +489,7 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         mAdapter = adapter;
     }
 
-    /**
-     * Dismisses the app menu and cancels the drag-to-scroll if it is taking place.
-     */
+    /** Dismisses the app menu and cancels the drag-to-scroll if it is taking place. */
     void dismiss() {
         if (isShowing()) {
             mPopup.dismiss();
@@ -518,54 +551,71 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         return null;
     }
 
-    /**
-     * Invalidate the app menu data. See {@link AppMenuAdapter#notifyDataSetChanged}.
-     */
+    /** Invalidate the app menu data. See {@link AppMenuAdapter#notifyDataSetChanged}. */
     void invalidate() {
         if (mAdapter != null) mAdapter.notifyDataSetChanged();
     }
 
-    private int setMenuHeight(List<Integer> menuItemIds, List<Integer> heightList,
-            Rect appDimensions, int screenHeight, Rect padding, int footerHeight, int headerHeight,
-            View anchorView, @IdRes int groupDividerResourceId) {
-        anchorView.getLocationOnScreen(mTempLocation);
-        int anchorViewY = mTempLocation[1] - appDimensions.top;
-
+    private void setMenuHeight(
+            List<Integer> menuItemIds,
+            List<Integer> heightList,
+            Rect appDimensions,
+            Rect padding,
+            int footerHeight,
+            int headerHeight,
+            View anchorView,
+            @IdRes int groupDividerResourceId,
+            int anchorViewOffset) {
         int anchorViewImpactHeight = mIsByPermanentButton ? anchorView.getHeight() : 0;
 
-        // Set appDimensions.height() for abnormal anchorViewLocation.
-        if (anchorViewY > screenHeight) {
-            anchorViewY = appDimensions.height();
-        }
-        int availableScreenSpace = Math.max(
-                anchorViewY, appDimensions.height() - anchorViewY - anchorViewImpactHeight);
+        int availableScreenSpace =
+                appDimensions.height()
+                        - anchorViewOffset
+                        - padding.bottom
+                        - footerHeight
+                        - headerHeight
+                        - anchorViewImpactHeight;
 
-        availableScreenSpace -= (padding.bottom + footerHeight + headerHeight);
         if (mIsByPermanentButton) availableScreenSpace -= padding.top;
         if (availableScreenSpace <= 0 && sExceptionReporter != null) {
-            String logMessage = String.format(
+            String logMessage =
                     "there is no screen space for app menn, mIsByPermanentButton = "
-                    + mIsByPermanentButton + ", anchorViewY = " + anchorViewY
-                    + ", appDimensions.height() = " + appDimensions.height()
-                    + ", anchorView.getHeight() = " + anchorView.getHeight()
-                    + " padding.top = " + padding.top + ", padding.bottom = " + padding.bottom
-                    + ", footerHeight = " + footerHeight + ", headerHeight = " + headerHeight);
-            PostTask.postTask(TaskTraits.BEST_EFFORT_MAY_BLOCK,
+                            + mIsByPermanentButton
+                            + ", anchorViewOffset = "
+                            + anchorViewOffset
+                            + ", appDimensions.height() = "
+                            + appDimensions.height()
+                            + ", anchorView.getHeight() = "
+                            + anchorView.getHeight()
+                            + " padding.top = "
+                            + padding.top
+                            + ", padding.bottom = "
+                            + padding.bottom
+                            + ", footerHeight = "
+                            + footerHeight
+                            + ", headerHeight = "
+                            + headerHeight;
+            PostTask.postTask(
+                    TaskTraits.BEST_EFFORT_MAY_BLOCK,
                     () -> sExceptionReporter.onResult(new Throwable(logMessage)));
         }
 
-        int menuHeight = calculateHeightForItems(
-                menuItemIds, heightList, groupDividerResourceId, availableScreenSpace);
+        int menuHeight =
+                calculateHeightForItems(
+                        menuItemIds, heightList, groupDividerResourceId, availableScreenSpace);
         menuHeight += footerHeight + headerHeight + padding.top + padding.bottom;
         mPopup.setHeight(menuHeight);
-        return menuHeight;
     }
 
     @VisibleForTesting
-    int calculateHeightForItems(List<Integer> menuItemIds, List<Integer> heightList,
-            @IdRes int groupDividerResourceId, int screenSpaceForItems) {
+    int calculateHeightForItems(
+            List<Integer> menuItemIds,
+            List<Integer> heightList,
+            @IdRes int groupDividerResourceId,
+            int screenSpaceForItems) {
         int availableScreenSpace = screenSpaceForItems > 0 ? screenSpaceForItems : 0;
         int spaceForFullItems = 0;
+
         for (int i = 0; i < heightList.size(); i++) {
             spaceForFullItems += heightList.get(i);
         }
@@ -591,6 +641,11 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
             while (lastItem > 1
                     && (spaceForItems + spaceForPartialItem > availableScreenSpace
                             || menuItemIds.get(lastItem) == groupDividerResourceId)) {
+                // If we have space for < 2.5 items, size menu to available screen space.
+                if (spaceForItems <= availableScreenSpace && lastItem < 3) {
+                    spaceForPartialItem = availableScreenSpace - spaceForItems;
+                    break;
+                }
                 spaceForItems -= heightList.get(lastItem - 1);
                 spaceForPartialItem =
                         (int) (LAST_ITEM_SHOW_FRACTION * heightList.get(lastItem - 1));
@@ -646,8 +701,9 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
     private int inflateHeader(int headerResourceId, View contentView, int menuWidth) {
         if (headerResourceId == 0) return 0;
 
-        View headerView = LayoutInflater.from(contentView.getContext())
-                                  .inflate(headerResourceId, mListView, false);
+        View headerView =
+                LayoutInflater.from(contentView.getContext())
+                        .inflate(headerResourceId, mListView, false);
         mListView.addHeaderView(headerView);
 
         int widthMeasureSpec = MeasureSpec.makeMeasureSpec(menuWidth, MeasureSpec.EXACTLY);
@@ -659,16 +715,16 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         return headerView.getMeasuredHeight();
     }
 
-    @VisibleForTesting
     void finishAnimationsForTests() {
         if (mMenuItemEnterAnimator != null) mMenuItemEnterAnimator.end();
     }
 
     private void recordTimeToTakeActionHistogram() {
-        final String histogramName = "Mobile.AppMenu.TimeToTakeAction."
-                + (mSelectedItemBeforeDismiss ? "SelectedItem" : "Abandoned");
+        final String histogramName =
+                "Mobile.AppMenu.TimeToTakeAction."
+                        + (mSelectedItemBeforeDismiss ? "SelectedItem" : "Abandoned");
         final long timeToTakeActionMs = SystemClock.elapsedRealtime() - mMenuShownTimeMs;
-        RecordHistogram.recordMediumTimesHistogram(histogramName, timeToTakeActionMs);
+        RecordHistogram.deprecatedRecordMediumTimesHistogram(histogramName, timeToTakeActionMs);
     }
 
     private int getMenuItemHeight(
@@ -685,7 +741,9 @@ class AppMenu implements OnItemClickListener, OnKeyListener, AppMenuClickHandler
         return mItemRowHeight;
     }
 
-    /** @param reporter A means of reporting an exception without crashing. */
+    /**
+     * @param reporter A means of reporting an exception without crashing.
+     */
     static void setExceptionReporter(Callback<Throwable> reporter) {
         sExceptionReporter = reporter;
     }
