diff --git a/base/values.cc b/base/values.cc
index 83f941d3..950291bc 100644
--- a/base/values.cc
+++ b/base/values.cc
@@ -1776,6 +1776,26 @@ std::unique_ptr<ListValue> ListValue::From(std::unique_ptr<Value> value) {
 
 ListValue::ListValue() : Value(Type::LIST) {}
 
+bool ListValue::GetDictionary(size_t index,
+                              const DictionaryValue** out_value) const {
+  const auto& list = GetListDeprecated();
+  if (list.size() <= index)
+    return false;
+  const base::Value& value = list[index];
+  if (!value.is_dict())
+    return false;
+
+  if (out_value)
+    *out_value = static_cast<const DictionaryValue*>(&value);
+
+  return true;
+}
+
+bool ListValue::GetDictionary(size_t index, DictionaryValue** out_value) {
+  return as_const(*this).GetDictionary(
+      index, const_cast<const DictionaryValue**>(out_value));
+}
+
 void ListValue::Append(base::Value::Dict in_dict) {
   list().emplace_back(std::move(in_dict));
 }
