diff --git a/base/android/java/src/org/chromium/base/SysUtils.java b/base/android/java/src/org/chromium/base/SysUtils.java
index ab5ef156..07cc3d5a 100644
--- a/base/android/java/src/org/chromium/base/SysUtils.java
+++ b/base/android/java/src/org/chromium/base/SysUtils.java
@@ -6,6 +6,8 @@ package org.chromium.base;
 
 import android.app.ActivityManager;
 import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager.NameNotFoundException;
 import android.content.pm.PackageManager;
 import android.os.Build;
 import android.os.Environment;
@@ -110,6 +112,30 @@ public class SysUtils {
         return 0;
     }
 
+    @CalledByNative
+    public static long firstInstallDate() {
+        Context context = ContextUtils.getApplicationContext();
+        if (context == null)
+            return 0;
+        PackageInfo packageInfo;
+        try {
+            packageInfo = context.getPackageManager().getPackageInfo(context.getPackageName(), 0);
+        } catch (NameNotFoundException e) {
+            packageInfo = null;
+        }
+        return packageInfo == null ? 0 : (packageInfo.firstInstallTime / 1000);
+    }
+
+    @CalledByNative
+    public static String referrerString() {
+        return ContextUtils.getAppSharedPreferences().getString("install_referrer", "");
+    }
+
+    @CalledByNative
+    public static String nightModeSettings() {
+        return ContextUtils.getAppSharedPreferences().getString("night_mode_settings", "");
+    }
+
     /**
      * @return Whether or not this device should be considered a low end device.
      */
