diff --git a/chrome/browser/download/chrome_download_manager_delegate.cc b/chrome/browser/download/chrome_download_manager_delegate.cc
index b517f633..7a9cbd9b 100644
--- a/chrome/browser/download/chrome_download_manager_delegate.cc
+++ b/chrome/browser/download/chrome_download_manager_delegate.cc
@@ -469,14 +469,15 @@ void ChromeDownloadManagerDelegate::ShowDownloadDialog(
     int64_t total_bytes,
     DownloadLocationDialogType dialog_type,
     const base::FilePath& suggested_path,
-    DownloadDialogBridge::DialogCallback callback) {
+    DownloadDialogBridge::DialogCallback callback,
+    download::DownloadItem* download) {
   DCHECK(download_dialog_bridge_);
   auto connection_type = net::NetworkChangeNotifier::GetConnectionType();
   bool is_incognito = profile_->IsOffTheRecord();
 
   download_dialog_bridge_->ShowDialog(
       native_window, total_bytes, connection_type, dialog_type, suggested_path,
-      is_incognito, std::move(callback));
+      is_incognito, std::move(callback), download);
 }
 
 void ChromeDownloadManagerDelegate::SetDownloadDialogBridgeForTesting(
@@ -1076,12 +1077,18 @@ void ChromeDownloadManagerDelegate::RequestConfirmation(
         return;
       }
 
+      if (false)
       if (!download_prefs_->PromptForDownload()) {
         DuplicateDownloadDialogBridgeDelegate::GetInstance()->CreateDialog(
             download, suggested_path, web_contents, std::move(callback));
         return;
       }
 
+      // This should be passed as a callback parameter
+      // but we avoid doing so to limit the amount of code we modify
+      // and simplify further rebasing
+      // (Arnaud)
+      last_download_item_ = download;
       gfx::NativeWindow native_window = web_contents->GetTopLevelNativeWindow();
       DownloadPathReservationTracker::GetReservedPath(
           download, suggested_path, download_dir,
@@ -1119,7 +1126,7 @@ void ChromeDownloadManagerDelegate::RequestConfirmation(
     gfx::NativeWindow native_window = web_contents->GetTopLevelNativeWindow();
     ShowDownloadDialog(
         native_window, download->GetTotalBytes(), dialog_type, suggested_path,
-        base::BindOnce(&OnDownloadDialogClosed, std::move(callback)));
+        base::BindOnce(&OnDownloadDialogClosed, std::move(callback)), download);
     return;
 
 #else   // BUILDFLAG(IS_ANDROID)
@@ -1192,7 +1199,7 @@ void ChromeDownloadManagerDelegate::GenerateUniqueFileNameDone(
       ShowDownloadDialog(
           native_window, 0 /* total_bytes */,
           DownloadLocationDialogType::NAME_CONFLICT, target_path,
-          base::BindOnce(&OnDownloadDialogClosed, std::move(callback)));
+          base::BindOnce(&OnDownloadDialogClosed, std::move(callback)), last_download_item_);
       return;
     }
 
