diff --git a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
index 2a107a4c..d4f6fce5 100644
--- a/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/settings/MainSettings.java
@@ -183,6 +183,10 @@ public class MainSettings extends PreferenceFragmentCompat
             setManagedPreferenceDelegateForPreference(PREF_PASSWORDS);
         }
 
+        Preference safetyCheck = findPreference(PREF_SAFETY_CHECK);
+        if (safetyCheck != null)
+            getPreferenceScreen().removePreference(safetyCheck);
+
         setManagedPreferenceDelegateForPreference(PREF_SEARCH_ENGINE);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
@@ -397,6 +401,8 @@ public class MainSettings extends PreferenceFragmentCompat
     }
 
     private void onSyncPromoPreferenceStateChanged() {
+        if (true)
+            return;
         if (ChromeFeatureList.isEnabled(ChromeFeatureList.SYNC_ANDROID_PROMOS_WITH_ILLUSTRATION)
                 || ChromeFeatureList.isEnabled(
                         ChromeFeatureList.SYNC_ANDROID_PROMOS_WITH_SINGLE_BUTTON)
