diff --git a/chrome/browser/signin/ui/android/java/src/org/chromium/chrome/browser/signin/ui/SigninPromoUtil.java b/chrome/browser/signin/ui/android/java/src/org/chromium/chrome/browser/signin/ui/SigninPromoUtil.java
index ebbd6765..83936bf5 100644
--- a/chrome/browser/signin/ui/android/java/src/org/chromium/chrome/browser/signin/ui/SigninPromoUtil.java
+++ b/chrome/browser/signin/ui/android/java/src/org/chromium/chrome/browser/signin/ui/SigninPromoUtil.java
@@ -40,6 +40,8 @@ public final class SigninPromoUtil {
     public static boolean launchSigninPromoIfNeeded(Context context,
             SyncConsentActivityLauncher syncConsentActivityLauncher,
             final int currentMajorVersion) {
+        if (true)
+          return false;
         final SigninPreferencesManager prefManager = SigninPreferencesManager.getInstance();
         if (shouldLaunchSigninPromo(prefManager, currentMajorVersion)) {
             syncConsentActivityLauncher.launchActivityIfAllowed(
