diff --git a/chrome/browser/extensions/proxy_overridden_bubble_delegate.cc b/chrome/browser/extensions/proxy_overridden_bubble_delegate.cc
index 80c7cd8c..201588cb 100644
--- a/chrome/browser/extensions/proxy_overridden_bubble_delegate.cc
+++ b/chrome/browser/extensions/proxy_overridden_bubble_delegate.cc
@@ -43,6 +43,8 @@ ProxyOverriddenBubbleDelegate::~ProxyOverriddenBubbleDelegate() {}
 
 bool ProxyOverriddenBubbleDelegate::ShouldIncludeExtension(
     const Extension* extension) {
+  if (true)
+    return false;
   if (!extension_id_.empty() && extension_id_ != extension->id())
     return false;  // Only one extension can be controlling the proxy at a time.
 
