diff --git a/chrome/browser/search/instant_service.cc b/chrome/browser/search/instant_service.cc
index b8949948..0a84ac09 100644
--- a/chrome/browser/search/instant_service.cc
+++ b/chrome/browser/search/instant_service.cc
@@ -25,6 +25,7 @@
 #include "chrome/browser/search/instant_service_factory.h"
 #include "chrome/browser/search/instant_service_observer.h"
 #include "chrome/browser/search/most_visited_iframe_source.h"
+#include "chrome/browser/search/new_tab_page_source.h"
 #include "chrome/browser/search/search.h"
 #include "chrome/browser/themes/theme_properties.h"
 #include "chrome/browser/themes/theme_service.h"
@@ -94,6 +95,8 @@ InstantService::InstantService(Profile* profile)
                     profile_, chrome::FaviconUrlFormat::kFaviconLegacy));
   content::URLDataSource::Add(profile_,
                               std::make_unique<MostVisitedIframeSource>());
+  content::URLDataSource::Add(profile_,
+                              std::make_unique<NewTabPageSource>(profile_));
 
   theme_observation_.Observe(native_theme_.get());
 }
@@ -117,8 +120,11 @@ void InstantService::RemoveObserver(InstantServiceObserver* observer) {
 }
 
 void InstantService::OnNewTabPageOpened() {
+  LOG(INFO) << "[Kiwi] InstantService::OnNewTabPageOpened";
   if (most_visited_sites_) {
+    LOG(INFO) << "[Kiwi] InstantService::OnNewTabPageOpened - most_visited_sites_";
     most_visited_sites_->Refresh();
+    most_visited_sites_->RefreshTiles();
   }
 }
 
@@ -215,6 +221,7 @@ void InstantService::OnURLsAvailable(
   // Use only personalized tiles for instant service.
   const ntp_tiles::NTPTilesVector& tiles =
       sections.at(ntp_tiles::SectionType::PERSONALIZED);
+  LOG(INFO) << "[Kiwi] InstantService::OnURLsAvailable - tiles: " << tiles.size();
   for (const ntp_tiles::NTPTile& tile : tiles) {
     InstantMostVisitedItem item;
     item.url = tile.url;
@@ -229,6 +236,7 @@ void InstantService::OnURLsAvailable(
 void InstantService::OnIconMadeAvailable(const GURL& site_url) {}
 
 void InstantService::NotifyAboutMostVisitedInfo() {
+  LOG(INFO) << "[Kiwi] InstantService::NotifyAboutMostVisitedInfo";
   for (InstantServiceObserver& observer : observers_)
     observer.MostVisitedInfoChanged(*most_visited_info_);
 }
