diff --git a/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/v2/NativeViewListRenderer.java b/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/v2/NativeViewListRenderer.java
index 9d6980bd..58fd3566 100644
--- a/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/v2/NativeViewListRenderer.java
+++ b/chrome/android/feed/core/java/src/org/chromium/chrome/browser/feed/v2/NativeViewListRenderer.java
@@ -120,6 +120,7 @@ public class NativeViewListRenderer extends RecyclerView.Adapter<NativeViewListR
 
     private View createCannotRenderViewItem() {
         TextView viewItem = new TextView(mContext);
+if (false) {
         String message = "Unable to render external view";
         viewItem.setText(message);
 
@@ -128,7 +129,7 @@ public class NativeViewListRenderer extends RecyclerView.Adapter<NativeViewListR
         layoutParams.bottomMargin = ViewUtils.dpToPx(mContext, 25F);
         layoutParams.topMargin = ViewUtils.dpToPx(mContext, 25F);
         viewItem.setLayoutParams(layoutParams);
-
+}
         return viewItem;
     }
 }
