diff --git a/chrome/browser/extensions/extension_system_factory.cc b/chrome/browser/extensions/extension_system_factory.cc
index 7d88a6fe..75b722e0 100644
--- a/chrome/browser/extensions/extension_system_factory.cc
+++ b/chrome/browser/extensions/extension_system_factory.cc
@@ -92,8 +92,11 @@ ExtensionSystemFactory* ExtensionSystemFactory::GetInstance() {
 ExtensionSystemFactory::ExtensionSystemFactory()
     : ExtensionSystemProvider("ExtensionSystem",
                               BrowserContextDependencyManager::GetInstance()) {
+#if 0
   DCHECK(ExtensionsBrowserClient::Get())
       << "ExtensionSystemFactory must be initialized after BrowserProcess";
+#endif
+if (extensions::ExtensionsBrowserClient::Get())
   DependsOn(ExtensionSystemSharedFactory::GetInstance());
 }
 
