diff --git a/chrome/renderer/BUILD.gn b/chrome/renderer/BUILD.gn
index 0ad8a99a..a96fba8d 100644
--- a/chrome/renderer/BUILD.gn
+++ b/chrome/renderer/BUILD.gn
@@ -380,12 +380,12 @@ static_library("renderer") {
     ]
   }
 
-  if (!is_android) {
+  if (!is_android || is_android) {
     sources += [
       "accessibility/read_anything_app_controller.cc",
       "accessibility/read_anything_app_controller.h",
-      "media/chrome_speech_recognition_client.cc",
-      "media/chrome_speech_recognition_client.h",
+#      "media/chrome_speech_recognition_client.cc",
+#      "media/chrome_speech_recognition_client.h",
       "searchbox/searchbox.cc",
       "searchbox/searchbox.h",
       "searchbox/searchbox_extension.cc",
