diff --git a/chrome/browser/ui/browser_list.h b/chrome/browser/ui/browser_list.h
index 8f215d7b..a35c7d1f 100644
--- a/chrome/browser/ui/browser_list.h
+++ b/chrome/browser/ui/browser_list.h
@@ -15,9 +15,11 @@
 #include "base/observer_list.h"
 #include "build/build_config.h"
 
+#if 0
 #if BUILDFLAG(IS_ANDROID)
 #error This file should only be included on desktop.
 #endif
+#endif
 
 class Browser;
 class Profile;
