diff --git a/chrome/renderer/searchbox/searchbox_extension.cc b/chrome/renderer/searchbox/searchbox_extension.cc
index 570653de..bece1e4f 100644
--- a/chrome/renderer/searchbox/searchbox_extension.cc
+++ b/chrome/renderer/searchbox/searchbox_extension.cc
@@ -56,6 +56,8 @@
 #include "url/url_constants.h"
 #include "v8/include/v8.h"
 
+#include "chrome/renderer/chrome_render_thread_observer.h"
+
 namespace {
 
 const char kCSSBackgroundImageFormat[] = "-webkit-image-set("
@@ -289,9 +291,12 @@ content::RenderFrame* GetMainRenderFrameForCurrentContext() {
 }
 
 SearchBox* GetSearchBoxForCurrentContext() {
+  LOG(INFO) << "[Kiwi] SearchBox* GetSearchBoxForCurrentContext";
   content::RenderFrame* main_frame = GetMainRenderFrameForCurrentContext();
+  LOG(INFO) << "[Kiwi] SearchBox* GetSearchBoxForCurrentContext - main_frame: " << main_frame;
   if (!main_frame)
     return nullptr;
+  LOG(INFO) << "[Kiwi] SearchBox* GetSearchBoxForCurrentContext - searchbox: " << SearchBox::Get(main_frame);
   return SearchBox::Get(main_frame);
 }
 
@@ -461,6 +466,7 @@ class NewTabPageBindings : public gin::Wrappable<NewTabPageBindings> {
   static bool IsInputInProgress();
   static v8::Local<v8::Value> GetMostVisited(v8::Isolate* isolate);
   static bool GetMostVisitedAvailable(v8::Isolate* isolate);
+  static bool IsIncognito(v8::Isolate* isolate);
   static v8::Local<v8::Value> GetNtpTheme(v8::Isolate* isolate);
 
   // Handlers for JS functions visible to all NTPs.
@@ -489,6 +495,8 @@ gin::ObjectTemplateBuilder NewTabPageBindings::GetObjectTemplateBuilder(
       .SetProperty("mostVisited", &NewTabPageBindings::GetMostVisited)
       .SetProperty("mostVisitedAvailable",
                    &NewTabPageBindings::GetMostVisitedAvailable)
+      .SetProperty("isIncognito",
+                   &NewTabPageBindings::IsIncognito)
       .SetProperty("ntpTheme", &NewTabPageBindings::GetNtpTheme)
       // TODO(https://crbug.com/1020450): remove "themeBackgroundInfo" legacy
       // name when we're sure no third-party NTP needs it.
@@ -555,12 +563,20 @@ v8::Local<v8::Value> NewTabPageBindings::GetMostVisited(v8::Isolate* isolate) {
 // static
 bool NewTabPageBindings::GetMostVisitedAvailable(v8::Isolate* isolate) {
   const SearchBox* search_box = GetSearchBoxForCurrentContext();
+  if (search_box)
+    LOG(INFO) << "[Kiwi] NewTabPageBindings::GetMostVisitedAvailable - search_box available";
+  else
+    LOG(INFO) << "[Kiwi] NewTabPageBindings::GetMostVisitedAvailable - search_box NOT available";
   if (!search_box)
     return false;
 
   return search_box->AreMostVisitedItemsAvailable();
 }
 
+bool NewTabPageBindings::IsIncognito(v8::Isolate* isolate) {
+  return ChromeRenderThreadObserver::is_incognito_process();
+}
+
 // static
 v8::Local<v8::Value> NewTabPageBindings::GetNtpTheme(v8::Isolate* isolate) {
   const SearchBox* search_box = GetSearchBoxForCurrentContext();
@@ -614,7 +630,14 @@ v8::Local<v8::Value> NewTabPageBindings::GetMostVisitedItemData(
     v8::Isolate* isolate,
     int rid) {
   const SearchBox* search_box = GetSearchBoxForCurrentContext();
-  if (!search_box || !HasOrigin(GURL(chrome::kChromeSearchMostVisitedUrl)))
+  if (!search_box || 
+(
+!HasOrigin(GURL(chrome::kChromeSearchMostVisitedUrl))
+&&
+!HasOrigin(GURL("chrome-search://local-ntp/"))
+)
+
+)
     return v8::Null(isolate);
 
   InstantMostVisitedItem item;
