diff --git a/chrome/renderer/searchbox/searchbox.cc b/chrome/renderer/searchbox/searchbox.cc
index a0c35a0a..ea051f55 100644
--- a/chrome/renderer/searchbox/searchbox.cc
+++ b/chrome/renderer/searchbox/searchbox.cc
@@ -292,6 +292,7 @@ void SearchBox::FocusChanged(OmniboxFocusState new_focus_state,
 
 void SearchBox::MostVisitedInfoChanged(
     const InstantMostVisitedInfo& most_visited_info) {
+  LOG(INFO) << "[Kiwi] SearchBox::MostVisitedInfoChanged - Step 1";
   has_received_most_visited_ = true;
 
   std::vector<InstantMostVisitedItemIDPair> last_known_items;
@@ -301,8 +302,10 @@ void SearchBox::MostVisitedInfoChanged(
     return;  // Do not send duplicate onmostvisitedchange events.
   }
 
+  LOG(INFO) << "[Kiwi] SearchBox::MostVisitedInfoChanged - Step 2";
   most_visited_items_cache_.AddItems(most_visited_info.items);
   if (can_run_js_in_renderframe_) {
+    LOG(INFO) << "[Kiwi] SearchBox::MostVisitedInfoChanged - Dispatching";
     SearchBoxExtension::DispatchMostVisitedChanged(
         render_frame()->GetWebFrame());
   }
@@ -338,6 +341,7 @@ GURL SearchBox::GetURLForMostVisitedItem(InstantRestrictedID item_id) const {
 }
 
 void SearchBox::DidCommitProvisionalLoad(ui::PageTransition transition) {
+  LOG(INFO) << "[Kiwi] SearchBox::DidCommitProvisionalLoad";
   can_run_js_in_renderframe_ = true;
 }
 
