diff --git a/chrome/browser/extensions/extension_install_prompt.cc b/chrome/browser/extensions/extension_install_prompt.cc
index 339ae3f8..60b212ad 100644
--- a/chrome/browser/extensions/extension_install_prompt.cc
+++ b/chrome/browser/extensions/extension_install_prompt.cc
@@ -43,6 +43,9 @@
 #include "ui/gfx/image/image_skia.h"
 #include "ui/gfx/image/image_skia_rep.h"
 
+#include "content/public/browser/javascript_dialog_manager.h"
+#include "components/javascript_dialogs/app_modal_dialog_manager.h"
+
 using extensions::Extension;
 using extensions::Manifest;
 using extensions::PermissionMessage;
@@ -377,6 +380,17 @@ std::u16string ExtensionInstallPrompt::Prompt::GetUserCount() const {
   return std::u16string();
 }
 
+std::u16string ExtensionInstallPrompt::Prompt::GetPermissionsAsString() const {
+  std::u16string result = std::u16string();
+  result = result + GetDialogTitle() + u"\n\n";
+  result = result + GetPermissionsHeading() + u"\n\n";
+  result = result + base::JoinString(prompt_permissions_.permissions,
+                                     u"\n\n");
+  result = result + base::JoinString(prompt_permissions_.details,
+                                     u"\n");
+  return result;
+}
+
 size_t ExtensionInstallPrompt::Prompt::GetPermissionCount() const {
   return prompt_permissions_.permissions.size();
 }
@@ -506,6 +520,7 @@ ExtensionInstallPrompt::ExtensionInstallPrompt(content::WebContents* contents)
                    : nullptr),
       extension_(nullptr),
       install_ui_(extensions::CreateExtensionInstallUI(profile_)),
+      contents_(contents),
       show_params_(new ExtensionInstallPromptShowParams(contents)),
       did_call_show_dialog_(false) {}
 
@@ -611,7 +626,7 @@ void ExtensionInstallPrompt::OnImageLoaded(const gfx::Image& image) {
 void ExtensionInstallPrompt::LoadImageIfNeeded() {
   // Don't override an icon that was passed in. Also, |profile_| can be null in
   // unit tests.
-  if (!icon_.empty() || !profile_) {
+  if (!icon_.empty() || !profile_ || true) {
     ShowConfirmation();
     return;
   }
@@ -632,6 +647,33 @@ void ExtensionInstallPrompt::LoadImageIfNeeded() {
                                          weak_factory_.GetWeakPtr()));
 }
 
+// Ensures that OnDialogClosed is only called once.
+class CloseDialogCallbackWrapper
+    : public base::RefCountedThreadSafe<CloseDialogCallbackWrapper> {
+ public:
+
+  explicit CloseDialogCallbackWrapper(ExtensionInstallPrompt::DoneCallback callback)
+      : callback_(std::move(callback)) {}
+
+  void Run(bool dialog_was_suppressed,
+           bool success,
+           const std::u16string& user_input) {
+    if (success) {
+      LOG(INFO) << "[EXTENSIONS] We received result from extension dialog (ACCEPTED)";
+      std::move(callback_).Run(ExtensionInstallPrompt::DoneCallbackPayload(ExtensionInstallPrompt::Result::ACCEPTED));
+    } else {
+      LOG(INFO) << "[EXTENSIONS] We received result from extension dialog (REJECTED)";
+      std::move(callback_).Run(ExtensionInstallPrompt::DoneCallbackPayload(ExtensionInstallPrompt::Result::USER_CANCELED));
+    }
+  }
+
+ private:
+  friend class base::RefCountedThreadSafe<CloseDialogCallbackWrapper>;
+  ~CloseDialogCallbackWrapper() {}
+
+  ExtensionInstallPrompt::DoneCallback callback_;
+};
+
 void ExtensionInstallPrompt::ShowConfirmation() {
   std::unique_ptr<const PermissionSet> permissions_to_display;
 
@@ -665,6 +707,21 @@ void ExtensionInstallPrompt::ShowConfirmation() {
   // Notify observers.
   prompt_->OnDialogOpened();
 
+  if (contents_ && contents_->GetPrimaryMainFrame() != nullptr) {
+    LOG(INFO) << "[EXTENSIONS] contents_ is not empty, displaying prompt";
+    scoped_refptr<CloseDialogCallbackWrapper> wrapper = new CloseDialogCallbackWrapper(std::move(done_callback_));
+
+    if (permissions_to_display) {
+      bool ignored;
+      javascript_dialogs::AppModalDialogManager::GetInstance()->RunJavaScriptDialog(
+          contents_, contents_->GetPrimaryMainFrame(), content::JAVASCRIPT_DIALOG_TYPE_CONFIRM,
+          prompt_->GetPermissionsAsString(), std::u16string(),
+                   base::BindOnce(&CloseDialogCallbackWrapper::Run, wrapper, false),
+                   &ignored);
+      return;
+    }
+  }
+
   // If true, auto confirm is enabled and already handled the result.
   if (AutoConfirmPromptIfEnabled())
     return;
@@ -679,6 +736,17 @@ void ExtensionInstallPrompt::ShowConfirmation() {
 }
 
 bool ExtensionInstallPrompt::AutoConfirmPromptIfEnabled() {
+  // Arnaud, TO DO
+  if (true) {
+      base::ThreadTaskRunnerHandle::Get()->PostTask(
+          FROM_HERE,
+          base::BindOnce(
+              std::move(done_callback_),
+              DoneCallbackPayload(ExtensionInstallPrompt::Result::ACCEPTED,
+                                  extensions::ScopedTestDialogAutoConfirm::
+                                      GetJustification())));
+      return true;
+  }
   switch (extensions::ScopedTestDialogAutoConfirm::GetAutoConfirmValue()) {
     case extensions::ScopedTestDialogAutoConfirm::NONE:
       return false;
