diff --git a/chrome/android/chrome_java_sources.gni b/chrome/android/chrome_java_sources.gni
index 93a2c426..f9b13033 100644
--- a/chrome/android/chrome_java_sources.gni
+++ b/chrome/android/chrome_java_sources.gni
@@ -2,6 +2,29 @@
 # Use of this source code is governed by a BSD - style license that can be
 # found in the LICENSE file.
 
+kiwi_java_sources = [
+  "java/src/org/chromium/chrome/browser/AppMenuBridge.java",
+  "java/src/org/chromium/chrome/browser/PersonalizeResults.java",
+  "java/src/org/chromium/chrome/browser/FixDevToolsWindow.java",
+  "java/src/org/chromium/chrome/browser/accessibility/settings/NightScalePreference.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/StackLayout.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/StackLayoutBase.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/NonOverlappingStack.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/OverlappingStack.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/Stack.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/StackAnimation.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/StackTab.java",
+  "java/src/org/chromium/chrome/browser/compositor/layouts/phone/stack/StackViewAnimation.java",
+  "java/src/org/chromium/chrome/browser/translate/settings/TranslateSettings.java",
+  "java/src/org/chromium/chrome/browser/translate/settings/RadioButtonGroupTranslatePreference.java",
+  "java/src/org/chromium/chrome/browser/settings/TabSwitcherSettings.java",
+  "java/src/org/chromium/chrome/browser/settings/RadioButtonGroupTabSwitcherPreference.java",
+  "java/src/org/chromium/chrome/browser/settings/ToolbarSettings.java",
+  "java/src/org/chromium/chrome/browser/settings/NightModeSettings.java",
+  "java/src/org/chromium/chrome/browser/settings/RadioButtonGroupNightModePreference.java",
+  "java/src/org/chromium/chrome/browser/init/ChromeBrowserReferrer.java",
+]
+
 chrome_java_sources = [
   "java/src/com/google/android/apps/chrome/appwidget/bookmarks/BookmarkThumbnailWidgetProvider.java",
   "java/src/org/chromium/chrome/browser/ActivityTabProvider.java",
@@ -359,7 +382,6 @@ chrome_java_sources = [
   "java/src/org/chromium/chrome/browser/compositor/scene_layer/StaticTabSceneLayer.java",
   "java/src/org/chromium/chrome/browser/compositor/scene_layer/TabListSceneLayer.java",
   "java/src/org/chromium/chrome/browser/compositor/scene_layer/TabStripSceneLayer.java",
-  "java/src/org/chromium/chrome/browser/compositor/scene_layer/ToolbarSwipeSceneLayer.java",
   "java/src/org/chromium/chrome/browser/contacts_picker/ChromePickerAdapter.java",
   "java/src/org/chromium/chrome/browser/content/ContentUtils.java",
   "java/src/org/chromium/chrome/browser/content_capture/ContentCaptureHistoryDeletionObserver.java",
@@ -1220,3 +1242,5 @@ chrome_java_sources = [
   "java/src/org/chromium/chrome/browser/webauth/authenticator/CableAuthenticatorUSBActivity.java",
   "java/src/org/chromium/chrome/browser/webshare/ShareServiceImplementationFactory.java",
 ]
+
+chrome_java_sources += kiwi_java_sources
