diff --git a/chrome/browser/search/most_visited_iframe_source.cc b/chrome/browser/search/most_visited_iframe_source.cc
index d27385ce..b7d672fa 100644
--- a/chrome/browser/search/most_visited_iframe_source.cc
+++ b/chrome/browser/search/most_visited_iframe_source.cc
@@ -90,7 +90,9 @@ bool MostVisitedIframeSource::ShouldDenyXFrameOptions() {
 
 bool MostVisitedIframeSource::ServesPath(const std::string& path) const {
   return path == kTitleHTMLPath || path == kTitleCSSPath ||
-         path == kTitleJSPath;
+         path == kTitleJSPath ||
+         path == "/local-ntp.html" || path == "/local-ntp.js" ||
+         path == "/local-ntp.css" || path == "/new-ntp.html";
 }
 
 void MostVisitedIframeSource::SendResource(
