diff --git a/chrome/browser/ui/extensions/icon_with_badge_image_source.h b/chrome/browser/ui/extensions/icon_with_badge_image_source.h
index c27ce4b3..6c479f61 100644
--- a/chrome/browser/ui/extensions/icon_with_badge_image_source.h
+++ b/chrome/browser/ui/extensions/icon_with_badge_image_source.h
@@ -63,10 +63,10 @@ class IconWithBadgeImageSource : public gfx::CanvasImageSource {
     return paint_blocked_actions_decoration_;
   }
 
- private:
   // gfx::CanvasImageSource:
   void Draw(gfx::Canvas* canvas) override;
 
+ private:
   // Paints |badge_|, if any, on |canvas|.
   void PaintBadge(gfx::Canvas* canvas);
 
