diff --git a/components/browser_ui/widget/android/BUILD.gn b/components/browser_ui/widget/android/BUILD.gn
index 512d51a1..d8d1074b 100644
--- a/components/browser_ui/widget/android/BUILD.gn
+++ b/components/browser_ui/widget/android/BUILD.gn
@@ -90,6 +90,14 @@ android_library("java") {
     "java/src/org/chromium/components/browser_ui/widget/promo/PromoCardProperties.java",
     "java/src/org/chromium/components/browser_ui/widget/promo/PromoCardView.java",
     "java/src/org/chromium/components/browser_ui/widget/promo/PromoCardViewBinder.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/AdCardCoordinator.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/AdCardProperties.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/AdCardView.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/AdCardViewBinder.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/FeedCardCoordinator.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/FeedCardProperties.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/FeedCardView.java",
+    "java/src/org/chromium/components/browser_ui/widget/promo/FeedCardViewBinder.java",
     "java/src/org/chromium/components/browser_ui/widget/scrim/ScrimCoordinator.java",
     "java/src/org/chromium/components/browser_ui/widget/scrim/ScrimMediator.java",
     "java/src/org/chromium/components/browser_ui/widget/scrim/ScrimProperties.java",
@@ -133,8 +141,15 @@ android_library("java") {
     "//third_party/androidx:androidx_recyclerview_recyclerview_java",
     "//third_party/androidx:androidx_vectordrawable_vectordrawable_animated_java",
     "//third_party/androidx:androidx_vectordrawable_vectordrawable_java",
+    "//third_party/android_deps:google_firebase_firebase_analytics_java",
+    "//third_party/android_deps:org_checkerframework_checker_compat_qual_java",
+    "//third_party/android_deps:google_play_services_ads_lite_java",
+    "$google_play_services_package:google_play_services_ads_java",
+    "//third_party/androidx:androidx_constraintlayout_constraintlayout_java",
     "//ui/android:ui_java",
     "//url:gurl_java",
+    "//components/image_fetcher:java",
+    "//chrome/browser/profiles/android:java",
   ]
   resources_package = "org.chromium.components.browser_ui.widget"
 }
@@ -235,6 +250,8 @@ android_resources("java_resources") {
     "java/res/drawable/tile_view_highlight.xml",
     "java/res/drawable/tile_view_highlight_mask.xml",
     "java/res/drawable/tile_view_highlight_plain.xml",
+    "java/res/drawable/gnt_rounded_corners_shape.xml",
+    "java/res/drawable/gnt_outline_shape.xml",
     "java/res/layout/app_menu_divider.xml",
     "java/res/layout/app_menu_layout.xml",
     "java/res/layout/date_divided_adapter_header_view_holder.xml",
@@ -248,6 +265,11 @@ android_resources("java_resources") {
     "java/res/layout/modern_list_item_view_v2.xml",
     "java/res/layout/more_progress_button.xml",
     "java/res/layout/number_roll_view.xml",
+    "java/res/layout/feed_card_view_medium.xml",
+    "java/res/layout/feed_card_view_medium_placeholder.xml",
+    "java/res/layout/ad_card_view_large.xml",
+    "java/res/layout/ad_card_view_compact.xml",
+    "java/res/layout/ad_card_view_slim.xml",
     "java/res/layout/promo_card_view_compact.xml",
     "java/res/layout/promo_card_view_large.xml",
     "java/res/layout/promo_card_view_slim.xml",
