diff --git a/content/public/common/url_constants.h b/content/public/common/url_constants.h
index ebbb5796..3e24a63f 100644
--- a/content/public/common/url_constants.h
+++ b/content/public/common/url_constants.h
@@ -20,6 +20,7 @@ namespace content {
 CONTENT_EXPORT extern const char kChromeDevToolsScheme[];
 CONTENT_EXPORT extern const char kChromeErrorScheme[];
 CONTENT_EXPORT extern const char kChromeUIScheme[];  // Used for WebUIs.
+CONTENT_EXPORT extern const char kChromeSearchScheme[];  // Used for WebUIs.
 CONTENT_EXPORT extern const char kChromeUIUntrustedScheme[];
 CONTENT_EXPORT extern const char kGuestScheme[];
 CONTENT_EXPORT extern const char kViewSourceScheme[];
