diff --git a/extensions/browser/extension_registrar.cc b/extensions/browser/extension_registrar.cc
index 3321983f..1d2ee31c 100644
--- a/extensions/browser/extension_registrar.cc
+++ b/extensions/browser/extension_registrar.cc
@@ -35,8 +35,16 @@
 #include "chrome/browser/ash/crosapi/browser_util.h"
 #endif  // BUILDFLAG(IS_CHROMEOS_ASH)
 
+#include "chrome/browser/android/tab_android.h"
+#include "chrome/browser/ui/android/tab_model/tab_model.h"
+#include "chrome/browser/ui/android/tab_model/tab_model_list.h"
+#include "content/public/browser/web_contents.h"
+#include "chrome/browser/profiles/profile.h"
+
 using content::DevToolsAgentHost;
 
+std::string AppMenuBridge_GetRunningExtensionsInternal(Profile* profile, content::WebContents* web_contents);
+
 namespace extensions {
 
 ExtensionRegistrar::ExtensionRegistrar(content::BrowserContext* browser_context,
@@ -320,6 +328,8 @@ void ExtensionRegistrar::ReloadExtension(
     LoadErrorBehavior load_error_behavior) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
 
+  LOG(INFO) << "[EXTENSIONS] We are reloading extension: " << extension_id;
+
   base::FilePath path;
 
   const Extension* disabled_extension =
@@ -396,6 +406,8 @@ void ExtensionRegistrar::OnUnpackedExtensionReloadFailed(
 void ExtensionRegistrar::TerminateExtension(const ExtensionId& extension_id) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
 
+  const ExtensionId saved_extension_id = extension_id;
+  LOG(INFO) << "[EXTENSIONS] Calling ExtensionRegistrar::TerminateExtension on id: " << extension_id;
   scoped_refptr<const Extension> extension =
       registry_->enabled_extensions().GetByID(extension_id);
   if (!extension)
@@ -411,6 +423,7 @@ void ExtensionRegistrar::TerminateExtension(const ExtensionId& extension_id) {
   registry_->AddTerminated(extension);
   registry_->RemoveEnabled(extension_id);
   DeactivateExtension(extension.get(), UnloadedExtensionReason::TERMINATE);
+  ReloadExtension(saved_extension_id, LoadErrorBehavior::kQuiet);
 }
 
 void ExtensionRegistrar::UntrackTerminatedExtension(
@@ -505,6 +518,22 @@ void ExtensionRegistrar::ActivateExtension(const Extension* extension,
   // service worker-based, it may be necessary to spin up its context.
   if (BackgroundInfo::HasLazyContext(extension))
     MaybeSpinUpLazyContext(extension, is_newly_added);
+  AppMenuBridge_GetRunningExtensionsInternal(Profile::FromBrowserContext(browser_context_), nullptr);
+
+  TabModelList::TabModelVector tab_model_vector = TabModelList::models();
+  if (!tab_model_vector.size())
+    return ;
+
+  for (TabModel* tab_model : tab_model_vector) {
+    TabAndroid* tab = tab_model->GetTabAt(0);
+    if (!tab)
+      continue ;
+
+    if (tab->web_contents()) {
+      AppMenuBridge_GetRunningExtensionsInternal(Profile::FromBrowserContext(browser_context_), tab->web_contents());
+      continue ;
+    }
+  }
 }
 
 void ExtensionRegistrar::DeactivateExtension(const Extension* extension,
