diff --git a/extensions/common/manifest_handlers/default_locale_handler.cc b/extensions/common/manifest_handlers/default_locale_handler.cc
index de40adab..2de02f01 100644
--- a/extensions/common/manifest_handlers/default_locale_handler.cc
+++ b/extensions/common/manifest_handlers/default_locale_handler.cc
@@ -91,9 +91,11 @@ bool DefaultLocaleHandler::Validate(
 
   base::FilePath locale_path;
   while (!(locale_path = locales.Next()).empty()) {
+#if 0
     if (extension_l10n_util::ShouldSkipValidation(path, locale_path,
                                                   all_locales))
       continue;
+#endif
 
     base::FilePath messages_path = locale_path.Append(kMessagesFilename);
     base::FilePath gzipped_messages_path =
