diff --git a/extensions/common/user_script.cc b/extensions/common/user_script.cc
index a5b5b4f9..2d49638e 100644
--- a/extensions/common/user_script.cc
+++ b/extensions/common/user_script.cc
@@ -40,7 +40,7 @@ namespace extensions {
 
 // The bitmask for valid user script injectable schemes used by URLPattern.
 enum {
-  kValidUserScriptSchemes = URLPattern::SCHEME_CHROMEUI |
+  kValidUserScriptSchemes = URLPattern::SCHEME_CHROMEUI | URLPattern::SCHEME_CHROMESEARCH |
                             URLPattern::SCHEME_HTTP | URLPattern::SCHEME_HTTPS |
                             URLPattern::SCHEME_FILE | URLPattern::SCHEME_FTP |
                             URLPattern::SCHEME_UUID_IN_PACKAGE
@@ -74,6 +74,7 @@ int UserScript::ValidUserScriptSchemes(bool can_execute_script_everywhere) {
   if (!base::CommandLine::ForCurrentProcess()->HasSwitch(
           switches::kExtensionsOnChromeURLs)) {
     valid_schemes &= ~URLPattern::SCHEME_CHROMEUI;
+    valid_schemes &= ~URLPattern::SCHEME_CHROMESEARCH;
   }
   return valid_schemes;
 }
