diff --git a/third_party/blink/renderer/core/layout/text_autosizer.cc b/third_party/blink/renderer/core/layout/text_autosizer.cc
index 1786e8c8..edb24918 100644
--- a/third_party/blink/renderer/core/layout/text_autosizer.cc
+++ b/third_party/blink/renderer/core/layout/text_autosizer.cc
@@ -181,6 +181,8 @@ static bool BlockIsRowOfLinks(const LayoutBlock* block) {
 }
 
 static bool BlockHeightConstrained(const LayoutBlock* block) {
+  if (true)
+    return false;
   // FIXME: Propagate constrainedness down the tree, to avoid inefficiently
   // walking back up from each box.
   // FIXME: This code needs to take into account vertical writing modes.
