diff --git a/third_party/blink/renderer/core/page/create_window.cc b/third_party/blink/renderer/core/page/create_window.cc
index 09ad5629..2a6abc21 100644
--- a/third_party/blink/renderer/core/page/create_window.cc
+++ b/third_party/blink/renderer/core/page/create_window.cc
@@ -326,8 +326,25 @@ Frame* CreateNewWindow(LocalFrame& opener_frame,
   }
 
   // Sandboxed frames cannot open new auxiliary browsing contexts.
+  bool shouldBlockWindow = false;
   if (opener_window.IsSandboxed(
           network::mojom::blink::WebSandboxFlags::kPopups)) {
+    shouldBlockWindow = true;
+  }
+  if (opener_window.IsCrossSiteSubframe() && !(url.Host().Contains("google"))
+      && !(url.Host().Contains("paypal"))
+      && !(url.Host().Contains("pay"))
+      && !(url.Host().Contains("bank"))
+      && !(url.Host().Contains("id"))
+      && !(url.Host().Contains("ikano"))
+      && !(url.Host().Contains("klarna"))
+      && !(url.Host().Contains("pank"))
+      && !(url.Host().Contains("oauth"))
+      && !(url.Host().Contains("twitter"))
+      && !(url.Host().Contains("facebook"))
+      && !(url.Host().Contains("disqus")))
+    shouldBlockWindow = true;
+  if (shouldBlockWindow) {
     // FIXME: This message should be moved off the console once a solution to
     // https://bugs.webkit.org/show_bug.cgi?id=103274 exists.
     opener_window.AddConsoleMessage(MakeGarbageCollected<ConsoleMessage>(
