From 72736b6df9b4ad4340b72adb884b69096cfaa29c Mon Sep 17 00:00:00 2001
From: Arnaud GRANAL <arnaud@geometry.ee>
Date: Sat, 7 Aug 2021 00:34:07 +0000
Subject: [PATCH 173/506] Always disable can add to home screen for now

---
 .../chrome/browser/webapps/AddToHomescreenIPHController.java    | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/chrome/browser/webapps/android/java/src/org/chromium/chrome/browser/webapps/AddToHomescreenIPHController.java b/chrome/browser/webapps/android/java/src/org/chromium/chrome/browser/webapps/AddToHomescreenIPHController.java
index cdbd4ef8..dc8f3427 100644
--- a/chrome/browser/webapps/android/java/src/org/chromium/chrome/browser/webapps/AddToHomescreenIPHController.java
+++ b/chrome/browser/webapps/android/java/src/org/chromium/chrome/browser/webapps/AddToHomescreenIPHController.java
@@ -81,6 +81,8 @@ public class AddToHomescreenIPHController {
 
     private static boolean canShowAddToHomescreenMenuItem(Context context, Tab tab) {
         if (tab.isIncognito()) return false;
+        // Always disable can add to home screen for now, to-do add a ContextUtils check
+        if (true) return false;
 
         GURL url = tab.getUrl();
         if (url.isEmpty() || !url.isValid()) return false;
