From 2cc49e7c5f89cfbbcee352205234f02612d5dc42 Mon Sep 17 00:00:00 2001
From: Eduard Ursu <eduardursu78@gmail.com>
Date: Thu, 12 Aug 2021 00:10:25 +0200
Subject: [PATCH 318/506] Update icon_update_script.sh

---
 toolbox/icon_update_script.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/toolbox/icon_update_script.sh b/toolbox/icon_update_script.sh
index c55acb5f..c8dc2d3e 100644
--- a/toolbox/icon_update_script.sh
+++ b/toolbox/icon_update_script.sh
@@ -1,5 +1,5 @@
 
-find ./* -name $2 | while read filename
+find ~/ -name $2 | while read filename
 do
   width=`identify -format '%w' $filename`;
   height=`identify -format '%w' $filename`;
