From 8f404d05031012dc86389101f46d7ce81ca805b4 Mon Sep 17 00:00:00 2001
From: Kiwi Browser <40272275+kiwibrowser@users.noreply.github.com>
Date: Sat, 14 Aug 2021 02:07:58 +0300
Subject: [PATCH 389/506] Update linter.yml

---
 .github/workflows/linter.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/linter.yml b/.github/workflows/linter.yml
index 0abe25b1..3892ec4a 100644
--- a/.github/workflows/linter.yml
+++ b/.github/workflows/linter.yml
@@ -1,4 +1,4 @@
-name: Lint codebase
+name: Automatic: Lint codebase
 
 #############################
 # Start the job on all push #
