diff --git a/chrome/browser/ui/browser_commands.cc b/chrome/browser/ui/browser_commands.cc
index d86609ce..e7ebefd3 100644
--- a/chrome/browser/ui/browser_commands.cc
+++ b/chrome/browser/ui/browser_commands.cc
@@ -149,6 +149,8 @@
 #include "url/gurl.h"
 #include "url/url_constants.h"
 
+#include "components/signin/public/base/signin_metrics.h"
+
 #if BUILDFLAG(ENABLE_EXTENSIONS)
 #include "chrome/browser/ui/extensions/app_launch_params.h"
 #include "chrome/browser/ui/extensions/application_launch.h"
@@ -1305,12 +1307,14 @@ void GenerateQRCodeFromPageAction(Browser* browser) {
 }
 
 void SharingHubFromPageAction(Browser* browser) {
+#if 0
   WebContents* web_contents =
       browser->tab_strip_model()->GetActiveWebContents();
   sharing_hub::SharingHubBubbleController* controller =
       sharing_hub::SharingHubBubbleController::CreateOrGetFromWebContents(
           web_contents);
   controller->ShowBubble(share::ShareAttempt(web_contents));
+#endif
 }
 
 void ScreenshotCaptureFromPageAction(Browser* browser) {
