diff --git a/chrome/browser/extensions/extension_context_menu_model.cc b/chrome/browser/extensions/extension_context_menu_model.cc
index 0ccd4872..47354604 100644
--- a/chrome/browser/extensions/extension_context_menu_model.cc
+++ b/chrome/browser/extensions/extension_context_menu_model.cc
@@ -346,7 +346,10 @@ void ExtensionContextMenuModel::ExecuteCommand(int command_id,
     }
     case OPTIONS:
       DCHECK(OptionsPageInfo::HasOptionsPage(extension));
+#if 0
       ExtensionTabUtil::OpenOptionsPage(extension, browser_);
+#endif
+      ExtensionTabUtil::OpenOptionsPage(extension, GetActiveWebContents());
       break;
     case TOGGLE_VISIBILITY: {
       bool currently_visible = button_visibility_ == PINNED;
