diff --git a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
index b5f2331a..0968eddb 100644
--- a/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
+++ b/chrome/browser/ui/android/theme/java/src/org/chromium/chrome/browser/theme/ThemeUtils.java
@@ -48,6 +48,9 @@ public class ThemeUtils {
      */
     public static int getBackgroundColor(Tab tab) {
         if (tab.isNativePage()) return tab.getNativePage().getBackgroundColor();
+        boolean isIncognito = tab != null && tab.isIncognito();
+        boolean isLocalNTP = tab != null && tab.getUrl().getSpec().contains("/local-ntp/");
+        if (isIncognito && isLocalNTP) return Color.BLACK;
 
         WebContents tabWebContents = tab.getWebContents();
         RenderWidgetHostView rwhv =
