diff --git a/chrome/browser/ui/ui_features.h b/chrome/browser/ui/ui_features.h
index 2a177528..8289163b 100644
--- a/chrome/browser/ui/ui_features.h
+++ b/chrome/browser/ui/ui_features.h
@@ -38,7 +38,7 @@ extern const base::Feature kChromeWhatsNewUI;
 extern const base::Feature kChromeWhatsNewInMainMenuNewBadge;
 #endif
 
-#if !defined(ANDROID)
+#if !defined(ANDROID) || true
 extern const base::Feature kAccessCodeCastUI;
 #endif
 
