diff --git a/chrome/browser/startup_data.cc b/chrome/browser/startup_data.cc
index 7c783194..8655b627 100644
--- a/chrome/browser/startup_data.cc
+++ b/chrome/browser/startup_data.cc
@@ -149,7 +149,7 @@ StartupData::TakeProtoDatabaseProvider() {
 void StartupData::PreProfilePrefServiceInit() {
   pref_registry_ = base::MakeRefCounted<user_prefs::PrefRegistrySyncable>();
   ChromeBrowserMainExtraPartsProfiles::
-      EnsureBrowserContextKeyedServiceFactoriesBuilt();
+      EnsureBrowserContextKeyedServiceFactoriesBuilt(false);
 }
 
 void StartupData::CreateServicesInternal() {
