diff --git a/chrome/browser/ui/tab_helpers.h b/chrome/browser/ui/tab_helpers.h
index 413e357d..36b2a88b 100644
--- a/chrome/browser/ui/tab_helpers.h
+++ b/chrome/browser/ui/tab_helpers.h
@@ -19,7 +19,7 @@ class ChromeThinWebViewInitializer;
 }
 }  // namespace thin_webview
 
-#else
+// #else
 
 namespace chrome {
 class BrowserTabStripModelDelegate;
@@ -55,10 +55,10 @@ class TabHelpers {
 
   friend class TabAndroid;
   friend class android::TabWebContentsDelegateAndroid;
-#else
+
   friend class Browser;
   friend class chrome::BrowserTabStripModelDelegate;
-#endif  // BUILDFLAG(IS_ANDROID)
+#endif  // defined(OS_ANDROID)
 
   // chrome::Navigate creates WebContents that are destined for the tab strip,
   // and that might have WebUI that immediately calls back into random tab
