diff --git a/chrome/browser/download/android/download_dialog_bridge.cc b/chrome/browser/download/android/download_dialog_bridge.cc
index 9e284429..6254666b 100644
--- a/chrome/browser/download/android/download_dialog_bridge.cc
+++ b/chrome/browser/download/android/download_dialog_bridge.cc
@@ -16,6 +16,8 @@
 #include "components/prefs/pref_service.h"
 #include "ui/android/window_android.h"
 
+#include "components/download/public/common/download_item.h"
+
 // Default minimum file size in kilobyte to trigger download later feature.
 const int64_t kDownloadLaterDefaultMinFileSizeKb = 204800;
 
@@ -50,7 +52,8 @@ void DownloadDialogBridge::ShowDialog(
     DownloadLocationDialogType dialog_type,
     const base::FilePath& suggested_path,
     bool is_incognito,
-    DialogCallback dialog_callback) {
+    DialogCallback dialog_callback,
+    download::DownloadItem* download) {
   if (!native_window)
     return;
 
@@ -79,6 +82,10 @@ void DownloadDialogBridge::ShowDialog(
 
   is_dialog_showing_ = true;
 
+  std::string url_to_download = "";
+  if (download && !(download->GetURL().is_empty()))
+      url_to_download = download->GetURL().spec();
+
   JNIEnv* env = base::android::AttachCurrentThread();
   Java_DownloadDialogBridge_showDialog(
       env, java_obj_, native_window->GetJavaObject(),
@@ -86,7 +93,10 @@ void DownloadDialogBridge::ShowDialog(
       static_cast<int>(dialog_type),
       base::android::ConvertUTF8ToJavaString(env,
                                              suggested_path.AsUTF8Unsafe()),
-      false /*supports_later_dialog*/, is_incognito);
+      false /*supports_later_dialog*/, is_incognito,
+      base::android::ConvertUTF8ToJavaString(env,
+                                             url_to_download)
+  );
 }
 
 void DownloadDialogBridge::OnComplete(
