diff --git a/chrome/browser/search/search.h b/chrome/browser/search/search.h
index 1906394c..3893bb2e 100644
--- a/chrome/browser/search/search.h
+++ b/chrome/browser/search/search.h
@@ -44,7 +44,7 @@ bool IsInstantNTPURL(const GURL& url, Profile* profile);
 // Returns the New Tab page URL for the given |profile|.
 GURL GetNewTabPageURL(Profile* profile);
 
-#if !BUILDFLAG(IS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID) || true
 
 // Returns true if |url| should be rendered in the Instant renderer process.
 bool ShouldAssignURLToInstantRenderer(const GURL& url, Profile* profile);
