diff --git a/chrome/browser/browser_features.h b/chrome/browser/browser_features.h
index f860c19c..04a6e845 100644
--- a/chrome/browser/browser_features.h
+++ b/chrome/browser/browser_features.h
@@ -44,7 +44,7 @@ extern const base::Feature kDoubleTapToZoomInTabletMode;
 extern const base::Feature kEnableUniveralLinks;
 #endif
 
-#if !BUILDFLAG(IS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID) || true
 extern const base::Feature kCopyLinkToText;
 extern const base::Feature kMuteNotificationSnoozeAction;
 #endif
