diff --git a/chrome/browser/process_singleton.h b/chrome/browser/process_singleton.h
index 16bb3aa1..ad8c85de 100644
--- a/chrome/browser/process_singleton.h
+++ b/chrome/browser/process_singleton.h
@@ -20,7 +20,7 @@
 #include "base/process/process.h"
 #include "ui/gfx/native_widget_types.h"
 
-#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID) || true
 #include "base/files/scoped_temp_dir.h"
 #endif
 
@@ -130,7 +130,7 @@ class ProcessSingleton {
   // Clear any lock state during shutdown.
   void Cleanup();
 
-#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID) || true
   static void DisablePromptForTesting();
   static void SkipIsChromeProcessCheckForTesting(bool skip);
   static void SetUserOptedUnlockInUseProfileForTesting(bool set_unlock);
@@ -150,7 +150,7 @@ class ProcessSingleton {
   // On Windows, Create() has to be called before this.
   NotifyResult NotifyOtherProcess();
 
-#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID) || true
   // Exposed for testing.  We use a timeout on Linux, and in tests we want
   // this timeout to be short.
   NotifyResult NotifyOtherProcessWithTimeout(
@@ -179,7 +179,7 @@ class ProcessSingleton {
   HANDLE lock_file_;
   base::FilePath user_data_dir_;
   ShouldKillRemoteProcessCallback should_kill_remote_process_callback_;
-#elif BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID)
+#elif BUILDFLAG(IS_POSIX) && !BUILDFLAG(IS_ANDROID) || true
   // Return true if the given pid is one of our child processes.
   // Assumes that the current pid is the root of all pids of the current
   // instance.
