diff --git a/content/browser/BUILD.gn b/content/browser/BUILD.gn
index e7351dc8..34902e9b 100644
--- a/content/browser/BUILD.gn
+++ b/content/browser/BUILD.gn
@@ -2530,7 +2530,7 @@ source_set("browser") {
     ]
   }
 
-  if (!is_android) {
+  if (!is_android || true) {
     deps += [ "//content/browser/devtools:devtools_resources_extern" ]
   }
 
@@ -2976,7 +2976,8 @@ source_set("browser") {
       deps += [ "//gpu/vulkan/init" ]
     }
     libs += [ "jnigraphics" ]
-  } else {
+  }
+  if (true) {
     # Not Android.
     sources += [
       # The WebAuthn devtools protocol API is not supported in Android yet.
@@ -2988,7 +2989,7 @@ source_set("browser") {
       "devtools/devtools_frontend_host_impl.h",
       "devtools/protocol/webauthn_handler.cc",
       "devtools/protocol/webauthn_handler.h",
-      "media/session/audio_focus_delegate_default.cc",
+#      "media/session/audio_focus_delegate_default.cc",
 
       # The Direct Sockets API is not implemented on Android.
       "direct_sockets/direct_sockets_service_impl.cc",
