diff --git a/content/public/common/content_switches.cc b/content/public/common/content_switches.cc
index a60146c4..0c74c7e1 100644
--- a/content/public/common/content_switches.cc
+++ b/content/public/common/content_switches.cc
@@ -691,7 +691,7 @@ const char kSandboxIPCProcess[]             = "sandbox-ipc";
 
 // Enables shared array buffer on desktop, gated by an Enterprise Policy.
 // TODO(crbug.com/1144104) Remove when migration to COOP+COEP is complete.
-#if !BUILDFLAG(IS_ANDROID)
+#if !BUILDFLAG(IS_ANDROID) || true
 const char kSharedArrayBufferUnrestrictedAccessAllowed[] =
     "shared-array-buffer-unrestricted-access-allowed";
 #endif
