diff --git a/extensions/common/url_pattern.h b/extensions/common/url_pattern.h
index 026a096a..64c3b5b5 100644
--- a/extensions/common/url_pattern.h
+++ b/extensions/common/url_pattern.h
@@ -63,6 +63,7 @@ class URLPattern {
     SCHEME_WSS = 1 << 8,
     SCHEME_DATA = 1 << 9,
     SCHEME_UUID_IN_PACKAGE = 1 << 10,
+    SCHEME_CHROMESEARCH = 1 << 11,
 
     // IMPORTANT!
     // SCHEME_ALL will match every scheme, including chrome://, chrome-
