diff --git a/third_party/blink/renderer/core/paint/paint_auto_dark_mode.h b/third_party/blink/renderer/core/paint/paint_auto_dark_mode.h
index 9cc8acbe..9e84771c 100644
--- a/third_party/blink/renderer/core/paint/paint_auto_dark_mode.h
+++ b/third_party/blink/renderer/core/paint/paint_auto_dark_mode.h
@@ -23,15 +23,6 @@ inline AutoDarkMode PaintAutoDarkMode(DarkModeFilter::ElementRole role,
   return AutoDarkMode(role, auto_dark_mode_enabled);
 }
 
-inline AutoDarkMode BorderPaintAutoDarkMode(const ComputedStyle& style,
-                                            Color border_color) {
-  SkColor background_color =
-      style.VisitedDependentColor(GetCSSPropertyBackgroundColor()).Rgb();
-  return PaintAutoDarkMode(style,
-                           DarkModeFilter::DarkModeFilter::BorderElementRole(
-                               border_color.Rgb(), background_color));
-}
-
 class ImageClassifierHelper {
  public:
   CORE_EXPORT static ImageAutoDarkMode GetImageAutoDarkMode(
