diff --git a/third_party/blink/renderer/core/frame/local_frame_view.cc b/third_party/blink/renderer/core/frame/local_frame_view.cc
index d22a8596..12aacb78 100644
--- a/third_party/blink/renderer/core/frame/local_frame_view.cc
+++ b/third_party/blink/renderer/core/frame/local_frame_view.cc
@@ -1975,6 +1975,11 @@ Color LocalFrameView::DocumentBackgroundColor() {
   bool blend_with_base = true;
   LayoutObject* background_source = GetLayoutView();
 
+  auto* settings = frame_->GetSettings();
+  if (settings && settings->GetForceDarkModeEnabled()) {
+    return Color::kBlack;
+  }
+
   // If we have a fullscreen element grab the fullscreen color from the
   // backdrop.
   if (Document* doc = frame_->GetDocument()) {
