diff --git a/third_party/blink/renderer/core/paint/box_painter_base.cc b/third_party/blink/renderer/core/paint/box_painter_base.cc
index d5607b5b..d7899d97 100644
--- a/third_party/blink/renderer/core/paint/box_painter_base.cc
+++ b/third_party/blink/renderer/core/paint/box_painter_base.cc
@@ -139,7 +139,7 @@ void BoxPainterBase::PaintNormalBoxShadow(const PaintInfo& info,
                                           const ComputedStyle& style,
                                           PhysicalBoxSides sides_to_include,
                                           bool background_is_skipped) {
-  if (!style.BoxShadow())
+  if (!style.BoxShadow() || style.DarkColorScheme())
     return;
   GraphicsContext& context = info.context;
 
@@ -245,7 +245,7 @@ void BoxPainterBase::PaintInsetBoxShadowWithBorderRect(
     const PhysicalRect& border_rect,
     const ComputedStyle& style,
     PhysicalBoxSides sides_to_include) {
-  if (!style.BoxShadow())
+  if (!style.BoxShadow() || style.DarkColorScheme())
     return;
   auto bounds = RoundedBorderGeometry::PixelSnappedRoundedInnerBorder(
       style, border_rect, sides_to_include);
@@ -256,7 +256,7 @@ void BoxPainterBase::PaintInsetBoxShadowWithInnerRect(
     const PaintInfo& info,
     const PhysicalRect& inner_rect,
     const ComputedStyle& style) {
-  if (!style.BoxShadow())
+  if (!style.BoxShadow() || style.DarkColorScheme())
     return;
   auto bounds = RoundedBorderGeometry::PixelSnappedRoundedBorderWithOutsets(
       style, inner_rect, LayoutRectOutsets());
@@ -307,6 +307,8 @@ void BoxPainterBase::PaintInsetBoxShadow(const PaintInfo& info,
                                          const FloatRoundedRect& bounds,
                                          const ComputedStyle& style,
                                          PhysicalBoxSides sides_to_include) {
+  if (style.DarkColorScheme())
+    return;
   GraphicsContext& context = info.context;
 
   const ShadowList* shadow_list = style.BoxShadow();
