diff --git a/third_party/blink/renderer/core/css/css_gradient_value.cc b/third_party/blink/renderer/core/css/css_gradient_value.cc
index 23c2d843..0cf2c80e 100644
--- a/third_party/blink/renderer/core/css/css_gradient_value.cc
+++ b/third_party/blink/renderer/core/css/css_gradient_value.cc
@@ -52,6 +52,9 @@
 #include "third_party/blink/renderer/platform/wtf/text/wtf_string.h"
 #include "ui/gfx/geometry/size.h"
 
+#include "third_party/blink/renderer/core/frame/settings.h"
+#include "third_party/blink/renderer/core/frame/local_frame.h"
+
 namespace blink {
 namespace cssvalue {
 
@@ -122,6 +125,11 @@ scoped_refptr<Image> CSSGradientValue::GetImage(
   if (size.IsEmpty())
     return nullptr;
 
+  auto* settings = document.GetFrame()->GetSettings();
+  if (settings && settings->GetForceDarkModeEnabled()) {
+    return nullptr;
+  }
+
   if (is_cacheable_) {
     if (!Clients().Contains(&client))
       return nullptr;
