diff --git a/third_party/blink/renderer/core/paint/box_painter.cc b/third_party/blink/renderer/core/paint/box_painter.cc
index 4641fdb3..38a7d348 100644
--- a/third_party/blink/renderer/core/paint/box_painter.cc
+++ b/third_party/blink/renderer/core/paint/box_painter.cc
@@ -141,7 +141,7 @@ void BoxPainter::PaintBoxDecorationBackgroundWithRect(
   // FIXME: Should eventually give the theme control over whether the box
   // shadow should paint, since controls could have custom shadows of their
   // own.
-  if (box_decoration_data.ShouldPaintShadow()) {
+  if (box_decoration_data.ShouldPaintShadow() && !style.DarkColorScheme()) {
     BoxPainterBase::PaintNormalBoxShadow(
         paint_info, paint_rect, style, PhysicalBoxSides(),
         !box_decoration_data.ShouldPaintBackground());
@@ -182,7 +182,7 @@ void BoxPainter::PaintBoxDecorationBackgroundWithRect(
     }
   }
 
-  if (box_decoration_data.ShouldPaintShadow()) {
+  if (box_decoration_data.ShouldPaintShadow() && !style.DarkColorScheme()) {
     BoxPainterBase::PaintInsetBoxShadowWithBorderRect(paint_info, paint_rect,
                                                       style);
   }
@@ -196,7 +196,7 @@ void BoxPainter::PaintBoxDecorationBackgroundWithRect(
           !theme_painter.PaintBorderOnly(layout_box_.GetNode(), style,
                                          paint_info, snapped_paint_rect);
     }
-    if (!theme_painted) {
+    if (!theme_painted && !style.DarkColorScheme()) {
       BoxPainterBase::PaintBorder(
           layout_box_, layout_box_.GetDocument(), layout_box_.GeneratingNode(),
           paint_info, paint_rect, style,
