diff --git a/ui/webui/resources/cr_elements/find_shortcut_behavior.js b/ui/webui/resources/cr_elements/find_shortcut_behavior.js
index 393a7e97..fee963c9 100644
--- a/ui/webui/resources/cr_elements/find_shortcut_behavior.js
+++ b/ui/webui/resources/cr_elements/find_shortcut_behavior.js
@@ -4,7 +4,7 @@
 
 import {assert, assertNotReached} from '../js/assert.m.js';
 import {isMac} from '../js/cr.m.js';
-import {KeyboardShortcutList} from '../js/cr/ui/keyboard_shortcut_list.m.js';
+import {KeyboardShortcutList} from '../js/cr/ui/keyboard_shortcut_list.js';
 import {isTextInputElement} from '../js/util.m.js';
 
 /**
