diff --git a/ui/webui/resources/cr_elements/find_shortcut_mixin.ts b/ui/webui/resources/cr_elements/find_shortcut_mixin.ts
index 1a488879..d6e435bc 100644
--- a/ui/webui/resources/cr_elements/find_shortcut_mixin.ts
+++ b/ui/webui/resources/cr_elements/find_shortcut_mixin.ts
@@ -6,7 +6,7 @@ import {dedupingMixin, PolymerElement} from 'chrome://resources/polymer/v3_0/pol
 
 import {assert, assertNotReached} from '../js/assert.m.js';
 import {isMac} from '../js/cr.m.js';
-import {KeyboardShortcutList} from '../js/cr/ui/keyboard_shortcut_list.m.js';
+import {KeyboardShortcutList} from '../js/cr/ui/keyboard_shortcut_list.js';
 import {isTextInputElement} from '../js/util.m.js';
 
 /**
